---
title: Mucin Supplementation Analysis Report
author: "G. Brett Moreau"
date: "January 31, 2021"
output:
  html_document:
    toc: yes
---
``` {r setup, include = FALSE}
#### PACKAGES ################################################
#if (!requireNamespace("BiocManager", quietly = TRUE))
#  install.packages("BiocManager")
#BiocManager::install()
packageVersion("BiocManager") # I'm using version 1.30.16

#BiocManager::install("phyloseq")
library(phyloseq)
packageVersion("phyloseq") # I'm using version 1.38.0

#install.packages("tidyverse")
library(tidyverse)
packageVersion("tidyverse") # I'm using version 1.3.1

#install.packages("ggplot2")
library(ggplot2)
packageVersion("ggplot2") # I'm using version 3.3.5

#BiocManager::install("microbiome")
library(microbiome) 
packageVersion("microbiome") # I'm using version 1.16.0

#install.packages("vegan")
library(vegan)
packageVersion("vegan") # I'm using version 2.5.7

#install.packages("caret")
library(caret)
packageVersion("caret") # I'm using version 6.0.90

#install.packages("randomForest")
library(randomForest)
packageVersion("randomForest") # I'm using version 4.6.14

#install.packages("pROC")
library(pROC)
packageVersion("pROC") # I'm using version 1.18.0
```
### Summary of Results

* There were no differences in richness between samples, but evenness was increased in 4 week samples (both Stress Alone and Stress + Mucin) compared to earlier samples.

* Differences in beta diversity were observed between all groups, particularly between samples collected at four weeks and samples collected at earlier timepoints.

* Phylum- and Family-level community composition changes were observed across groups. Several Families (including *Lactobacillaceae* and *Muribaculaceae*) were changed between Baseline and 3 week stress conditions, while a couple (*Rikenellaceae* and *Marinifilaceae*) were changed between Stress Alone and Stress + Mucin conditions.

* Random Forest models successfully distinguish between both Baseline and 3 Week Stress groups and Stress Alone and Stress + Mucin Groups. 

<br>

### Introduction
The purpose of this experiment is to investigate the microbiome of stressed mice whose diet has or has not been supplemented  with mucin. Behavioral experiments have observed some changes in stress behavior after mucin supplementation. Previous experiments have also identified changes in the microbiota, particularly *Lactobacillus* spp., that are associated with changes in stress response. Because of this, we hypothesize that mucin supplementation influences the intestinal microbiota to drive behavioral changes.

<br>

### Characterization of the Data Set
There are 71 samples in this experiment. Samples were divided into two groups (Control and Mucin-supplemented) and three different timepoints (pre-stress, 3 weeks post-stress, and 4 weeks post-stress). Four week samples are broken into two groups: Stress Alone, which is the control group, and Stress + Mucin, which is has a diet supplemented with 5% mucin.  There are 24 mice at Baseline and 3 Weeks Stress,  12 mice for Stress Alone, and 11 mice for Stress + Mucin (Sample #167 was removed due to low read counts), giving our 71 samples.
<br>

Below is a general overview of the total number of reads in the data set, the mean and median number of reads/sample, and the range of reads across this data set.
<br>

```{r include = FALSE}
load(file = "/Users/gbm5pn/Documents/GitHub/Gautier-TUMI-Pilot-16S/results/phyloseq objects by experiment.RData")

ps.mucin.prop <- transform_sample_counts(ps.mucin, function(ASV) ASV/sum(ASV)) 

total.reads <- sample_sums(ps.mucin)
```

```{r}
sum(total.reads)
mean(total.reads) 
median(total.reads)
range(total.reads) 
```

<br>

### Alpha Diversity 
I'll first investigate alpha diversity, which examines diversity within a given sample. Alpha diversity consists of two major measures:

* **Richness**: the number of taxonomic groups within a given sample.

* **Evenness**: how evenly abundances of taxonomic groups are distributed in a given sample.
<br>

The DADA2 pipeline removes singlets (sequences detected only once) from the data set, as it does not have enough confidence to determine whether these sequences are real or artifacts. Because of this, measures of richness fail explicitly and are difficult to interpret. I'll focus mainly on measures of evenness but give measures of richness a glance to see if there are any major differences.
<br><br>

#### Richness
```{r echo = FALSE}
plot_richness(ps.mucin, x = "Group", measures = c("Observed"), 
              color = "Group") + 
  geom_point() +
  geom_boxplot() + 
  theme_bw() +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_blank()) +
  scale_color_discrete(name = "Group", 
                       breaks = c("Baseline", "3 Wk Stress", "Stress Alone", "Stress + Mucin"), 
                       labels = c("Baseline", "3 Wk Stress", "Stress Alone", "Stress + Mucin")) 


richness <- estimate_richness(ps.mucin)
pairwise.wilcox.test(richness$Observed, p.adjust.method = "bonferroni", sample_data(ps.mucin)$Group)

```
<br>

There are no major differences in alpha diversity between groups.
<br><br>

#### Evenness
```{r echo = FALSE}
# First I'll organize the data.
sample.data.mucin <- ps.mucin@sam_data
sample.data.mucin <- cbind("Sample.Names" = rownames(sample.data.mucin), sample.data.mucin)

evenness<- evenness(ps.mucin, index = c("pielou"))
evenness<- cbind("Sample.Names" = rownames(evenness), evenness)

sample.data.mucin <- full_join(sample.data.mucin, evenness, by = "Sample.Names")


### PIELOU ###
ggplot(sample.data.mucin, aes(x = Group, y = pielou, color = Group)) +
  geom_point() +
  geom_boxplot() +
  theme_bw() +
  labs(title = "Microbial Evenness Across Groups", x = NULL, y = "Pielou Eveness Index")
  theme(legend.position = "none")
```
<br>

Baseline and 3 Week Stress samples look similar to each other, while 4 week samples (Stress Alone and Stress + Mucin) appear to have higher evenness . A Pairwise Wilcoxon Rank Sum Test with Bonferroni correction from multiple comparisons was performed to test for differences between groups.
<br>

```{r echo = FALSE}
pairwise.wilcox.test(evenness$pielou, p.adjust.method = "bonferroni", sample_data(ps.mucin)$Group)
```
<br>

There was no significant difference in evenness between Baseline and 3 Week Stress samples. However, both 4 week samples (Stress Alone and Stress + Mucin) were significantly higher than either Baseline or 3 Week Stress groups. No significant difference in evenness was observed between Stress Alone and Stress + Mucin groups.

<br>

### Beta Diversity
I'll now investigate differences in beta diversity, or diversity between different samples in the experiment. I'll use Bray-Curtis Dissimilarity as my measure for beta diversity in this study, and I'll use Non-metric Multidimensional Scaling (NMDS) to visualize dissimilarity distances between samples.

``` {r include = FALSE}
#sample(1:1000, 1) # It selected 966
set.seed(966) # Set seed for reproducibility

# First, let's ordinate Bray-Curtis Dissimilarity using NMDS, then plot this ordination to visualize
# separation between different samples.

ord.nmds.bray <- ordinate(ps.mucin.prop, method = "NMDS", distance = "bray")
```

```{r echo = FALSE}
plot_ordination(ps.mucin.prop, ord.nmds.bray, color = "Group", 
                title = "Bray-Curtis Dissimilarity") +
  theme_bw() +
  scale_color_manual(name = "Group", 
                     values = c("#941650", "#1c7798", "#521b92", "#4d8e00"),
                       breaks = c("Baseline", "3 Wk Stress", "Stress Alone", "Stress + Mucin"), 
                       labels = c("Baseline", "3 Wk Stress", "Stress Alone", "Stress + Mucin")) +
  theme(aspect.ratio = 1, plot.title = element_text(hjust = 0.5)) 
```
<br>

Looking at the clustering, it appears that groups cluster distinctly from each other. Baseline and 3 Week Stress groups are relatively similar to each other, and 4 week samples (Stress Alone and Stress + Mucin) are relatively similar to each other. Baseline and 3 Week samples are quite distinct from 4 week samples.
<br>

I'll now use ADONIS to calculate statistical differences between groups.
<br>

```{r include = FALSE}
adonis(distance(ps.mucin.prop, method = "bray") ~sample_data(ps.mucin.prop)$Group)

# The PERMANOVA found significant differences in clustering (p = 0.001), but this doesn't specifiy
# which groups are significantly different from each other. To test this, I'll perform pairwise
# PERMANOVAs with multiple comparisons correction.

permanova.combinations <- combn(x = levels(ps.mucin.prop@sam_data$Group), m = 2)
permanova.p.bray <- vector(length = 6)
for(i in 1:ncol(permanova.combinations)){
  ps.subs.bray <- subset_samples(ps.mucin.prop, Group %in% permanova.combinations[,i])
  metadata.subs.bray <- as(sample_data(ps.subs.bray), "data.frame")
  pairwise.bray <- adonis(distance(ps.subs.bray, method = "bray") ~ Group, data = metadata.subs.bray)
  permanova.p.bray[i] <- pairwise.bray$aov.tab[1,ncol(pairwise.bray$aov.tab)]
}
combinations <- c("Baseline/3 Wk Stress", "Baseline/Stress Alone", "Baseline/Stress + Mucin", 
                  "3 Wk Stress/Stress Alone", "3 Wk Stress/Stress + Mucin",
                  "Stress Alone/Stress + Mucin")

permanova.FDR.bray <- p.adjust(permanova.p.bray, method = "bonferroni")
permanova.bray.table <- cbind(combinations, permanova.p.bray)
permanova.bray.table <- cbind(permanova.bray.table, permanova.FDR.bray)

```

```{r echo = FALSE}
library(knitr)
kable(permanova.bray.table)
```
<br>
The PERMANOVA data confirms the initial impressions from the NMDS plot: All groups cluster separately from each other and are statistically significant from every other group.

<br>

### Changes in Community Composition
There appear to be some broad differences in diversity between groups. I'll now see if this is reflected by any broad changes in community composition across groups. I'll start by looking at the phylum level, then move on to differences at the Family level. This analysis will focus on taxa that make up at least 1% of the relative abundance in at least one group.

```{r include = FALSE}
ps.mucin.prop.phylum <- tax_glom(ps.mucin.prop, "Phylum", NArm = FALSE)
phylum.table.mucin <- psmelt(ps.mucin.prop.phylum) # Organize in long format for ggplot.

# There are 24 samples for "Baseline" and "3 Wk Stress" groups, 12
# samples for the "Stress Alone" group, and 11 samples for the "Stress + Mucin" group. I'll 
# separate these groups to generate relative abundances as a percentage by dividing by the total
# number of samples in each group.

phylum.table.mucin.baseline.3wk <- filter(phylum.table.mucin, Group == "Baseline" | Group == "3 Wk Stress")
phylum.table.mucin.baseline.3wk$Relative.Abundance <- NA
phylum.table.mucin.baseline.3wk$Relative.Abundance <- phylum.table.mucin.baseline.3wk$Abundance / 24 *100


phylum.table.stress <- filter(phylum.table.mucin, Group == "Stress Alone")
phylum.table.stressRelative.Abundance <- NA
phylum.table.stress$Relative.Abundance <- phylum.table.stress$Abundance / 12 *100


phylum.table.stress.mucin <- filter(phylum.table.mucin, Group == "Stress + Mucin")
phylum.table.stress.mucinRelative.Abundance <- NA
phylum.table.stress.mucin$Relative.Abundance <- phylum.table.stress.mucin$Abundance / 11 *100


phylum.table.mucin <- rbind(phylum.table.mucin.baseline.3wk, phylum.table.stress)
phylum.table.mucin <- rbind(phylum.table.mucin, phylum.table.stress.mucin)


phylum.table.mucin <- as.data.frame(phylum.table.mucin)


# I'll plot the data to get an overview of differences between groups.
ggplot(phylum.table.mucin, aes(x = Group, y = Relative.Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") 

# This figure shows that the ratio of Bacteroidota to Firmicutes increases over time. However, I 
# want to clean up the figure by removing all Phyla that make up less than 1% of the total in any
# group and placing them in their own category.


# First, I'll summarize relative abundance information per taxonomic group.
summarized.abundance.phylum <- phylum.table.mucin %>%
  group_by(Phylum, Group) %>%
  summarize(Total.Relative.Abundance = (sum(Relative.Abundance)))

summarized.abundance.phylum$Phylum <- as.character(replace_na(summarized.abundance.phylum$Phylum, replace = "Unidentified"))

# Now, I'll collect taxa names for taxa with relative abundance above 1% in any group. These
# names will be preserved in each group.
summarized.abundance.phylum.above1 <- filter(summarized.abundance.phylum, Total.Relative.Abundance >= 1.0)
summarized.abundance.phylum.above1$Phylum <- factor(summarized.abundance.phylum.above1$Phylum, 
                                                    levels = sort(unique(summarized.abundance.phylum.above1$Phylum)))
abundant.families <- as.character(unique(summarized.abundance.phylum.above1$Phylum)) 
table(abundant.families) # There are 3 Phyla that have a Total Relative Abundance above 1%

# Make new data frames either including or excluding abundant families.
summarized.abundance.phylum.no.remainder <- filter(summarized.abundance.phylum, 
                                                   Phylum %in% abundant.families)

# I'll now summarize the remainder samples that compose less than 1% of relative abundance per group.
summarized.abundance.phylum.remainders.only <- anti_join(summarized.abundance.phylum, summarized.abundance.phylum.no.remainder, by = "Phylum")

summarized.abundance.phylum.remainders.only.summary <- summarized.abundance.phylum.remainders.only %>%
  group_by(Group) %>%
  summarize(Total.Relative.Abundance = (sum(Total.Relative.Abundance))) %>%
  mutate(Phylum = "Phyla < 1%", .before = 1)


# Now I'll combine abundant families with the low abundance remainders to account for 100% of total 
# abundance. This data frame will be used to generate the community composition figure.

summarized.abundance.phylum.figure <- rbind(summarized.abundance.phylum.no.remainder, summarized.abundance.phylum.remainders.only.summary)
summarized.abundance.phylum.figure$Phylum <- factor(summarized.abundance.phylum.figure$Phylum, levels = c("Bacteroidota", "Firmicutes", "Verrucomicrobiota", "Phyla < 1%"))

```
#### Phylum-level Community Composition
```{r echo = FALSE}
ggplot(summarized.abundance.phylum.figure, aes(x = Group, y = Total.Relative.Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  labs(title = "Community Composition (Phylum Level)", x = NULL, y = "Relative Abundance (%)") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
```
<br>

At the Phylum level, samples are dominated by members of the *Bacteroidota* and *Firmicutes* phyla. There is a general trend of relative decreased abundance in *Firmicutes* and relative increased abundance of *Bacteroidota* over the course of the experiment.
<br><br>

```{r include = FALSE}
ps.mucin.prop.family <- tax_glom(ps.mucin.prop, "Family", NArm = FALSE)
family.table.mucin <- psmelt(ps.mucin.prop.family) # Organize in long format for ggplot.

table(ps.mucin@sam_data$Group) # There are 24 samples for "Baseline" and "3 Wk Stress" groups, 
# 12 samples for the "Stress Alone" group, and 11 samples for the "Stress + Mucin" group. I'll 
# separate these groups to generate relative abundances as a percentage by dividing by the total
# number of samples in each group.

family.table.mucin.baseline.3wk <- filter(family.table.mucin, Group == "Baseline" | Group == "3 Wk Stress")
family.table.mucin.baseline.3wk$Relative.Abundance <- NA
family.table.mucin.baseline.3wk$Relative.Abundance <- family.table.mucin.baseline.3wk$Abundance / 24 *100


family.table.stress <- filter(family.table.mucin, Group == "Stress Alone")
family.table.stressRelative.Abundance <- NA
family.table.stress$Relative.Abundance <- family.table.stress$Abundance / 12 *100


family.table.stress.mucin <- filter(family.table.mucin, Group == "Stress + Mucin")
family.table.stress.mucinRelative.Abundance <- NA
family.table.stress.mucin$Relative.Abundance <- family.table.stress.mucin$Abundance / 11 *100


family.table.mucin <- rbind(family.table.mucin.baseline.3wk, family.table.stress)
family.table.mucin <- rbind(family.table.mucin, family.table.stress.mucin)


family.table.mucin <- as.data.frame(family.table.mucin)

# I'll plot the data to get an overview of differences between groups.
ggplot(family.table.mucin, aes(x = Group, y = Relative.Abundance, fill = Family)) +
  geom_bar(stat = "identity") 

# This figure is pretty cluttered. To clean it up, I'll filter out all Families that make up less than 1
# % of total in any group and place them in their own category.


# First, I'll summarize relative abundance information per taxonomic group.
summarized.abundance.family <- family.table.mucin %>%
  group_by(Family, Group) %>%
  summarize(Total.Relative.Abundance = (sum(Relative.Abundance)))

summarized.abundance.family$Family <- as.character(replace_na(summarized.abundance.family$Family, replace = "Unidentified"))

# Now, I'll collect taxa names for taxa with relative abundance above 1% in any group. These
# names will be preserved in each group.
summarized.abundance.family.above1 <- filter(summarized.abundance.family, Total.Relative.Abundance >= 1.0)
summarized.abundance.family.above1$Family <- factor(summarized.abundance.family.above1$Family, 
                                                    levels = sort(unique(summarized.abundance.family.above1$Family)))
abundant.families <- as.character(unique(summarized.abundance.family.above1$Family)) 
table(abundant.families) # There are 13 Families (and 1 group of unidentified families) that have a Total Relative 
# Abundance above 1%

# Make new data frames either including or excluding abundant families.
summarized.abundance.family.no.remainder <- filter(summarized.abundance.family, 
                                                   Family %in% abundant.families)

# I'll now summarize the remainder samples that compose less than 1% of relative abundance per group.
summarized.abundance.family.remainders.only <- anti_join(summarized.abundance.family, summarized.abundance.family.no.remainder, by = "Family")

summarized.abundance.family.remainders.only.summary <- summarized.abundance.family.remainders.only %>%
  group_by(Group) %>%
  summarize(Total.Relative.Abundance = (sum(Total.Relative.Abundance))) %>%
  mutate(Family = "Families < 1%", .before = 1)


# Now I'll combine abundant families with the low abundance remainders to account for 100% of total 
# abundance. This data frame will be used to generate the community composition figure.

summarized.abundance.family.figure <- rbind(summarized.abundance.family.no.remainder, summarized.abundance.family.remainders.only.summary)
summarized.abundance.family.figure$Family <- factor(summarized.abundance.family.figure$Family, levels = c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", 
                                                                                                          "Lactobacillaceae", "Marinifilaceae", "Muribaculaceae", "Oscillospiraceae", "Peptostreptococcaceae", "Rikenellaceae", "Ruminococcaceae", 
                                                                                                          "Unidentified", "Families < 1%"))
```
<br>

#### Family-Level Community Composition

```{r echo = FALSE}

ggplot(summarized.abundance.family.figure, aes(x = Group, y = Total.Relative.Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  labs(title = "Community Composition (Family Level)", x = NULL, y = "Relative Abundance (%)") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
```

Several Families are abundant across all groups, with the most abundant families including *Lachnospiraceae* and *Muribaculaceae*. I'll directly compare relative abundances of these families between groups of interest (Baseline vs.3 Week Stress and Stress Alone vs. Stress + Mucin).
<br>

#### Baseline vs. 3 Week Stress


```{r include = FALSE}
#### FAMILY LEVEL DIFFERENCES: BASELINE VS 3 WEEK STRESS ############################################
# I now want to plot these differences by Family group. I'll only be plotting the abundant families
# (those with a relative abundance above 1% in at least one Group). I'll start by taking the family
# table I generated previously and filtering it to include only these abundant families.

family.table.mucin.abundant <- filter(family.table.mucin, Family %in% abundant.families)

# I'll look at these comparisons in pairs to keep the figure from getting too complicated. I'll start
# by looking at the effect of stress alone, looking at Baseline vs. 3 week stress samples. I'll
# filter to include only these samples.

family.table.mucin.abundant.stress <- filter(family.table.mucin.abundant, Group == "Baseline" | Group == "3 Wk Stress")
family.table.mucin.abundant.stress$Group <- factor(family.table.mucin.abundant.stress$Group, levels = c("Baseline", "3 Wk Stress"))

family.table.mucin.abundant.stress <- filter(family.table.mucin.abundant.stress, Family != "Marinifilaceae")
family.table.mucin.abundant.stress <- filter(family.table.mucin.abundant.stress, Family != "Rikenellaceae")
family.table.mucin.abundant.stress$Family <- factor(family.table.mucin.abundant.stress$Family, levels = c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", 
                                                                                                          "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", 
                                                                                                          "Muribaculaceae", "Oscillospiraceae", "Peptostreptococcaceae", "Ruminococcaceae"))
```

```{r echo = FALSE}
ggplot(family.table.mucin.abundant.stress, aes(x = Abundance, y = Family, color = Group)) +
  geom_boxplot() +
  theme_bw() +
  scale_x_continuous(labels = scales::percent) +
  scale_y_discrete(limits = rev) +
  scale_color_manual(name = "Group", 
                     values = c("#941650", "#1c7798"), 
                     breaks = c("Baseline", "3 Wk Stress"), 
                     labels = c("Baseline", "3 Wk Stress")) +
  labs(title = "Family Abundance (Baseline vs. 3 Week Stresss)", x = "Relative Abundance", y = NULL) +
  theme(plot.title = element_text(hjust = 0.5))
```


``` {r include = FALSE}
### SIGNIFICANT DIFFERENCES: BASELINE VS. 3 WK STRESS ###
# I'll now use a Wilcoxon rank test to look for significant differences between each of these Family 
# groups. First, I'll organize the data so that abundance per Family is summarized on a per sample 
# basis. I'll also get Sample and Group metadata to add to summarized tables.
wilcox.family<- family.table.mucin.abundant.stress %>%
  group_by(Family, Sample) %>%
  summarize(Abundance.Group = (sum(Abundance)))

Sample.Group <- select(family.table.mucin.abundant.stress, Sample, Group)
Sample.Group <- Sample.Group %>%
  group_by(Sample) %>%
  slice(1L)

wilcox.family <- full_join(wilcox.family, Sample.Group, by = "Sample")

# Now I'll calculate Wilcoxon Ranked Sum p values for each Family.

### EUBACTERIA ###
wilcox.eubac <- filter(wilcox.family, Family == "[Eubacterium] coprostanoligenes group")

wilcox.eubac.baseline <- filter(wilcox.eubac, Group == "Baseline")
wilcox.eubac.3wk <- filter(wilcox.eubac, Group == "3 Wk Stress")

wilcox.test.eubac <- wilcox.test(wilcox.eubac.baseline$Abundance.Group, 
                                 wilcox.eubac.3wk$Abundance.Group)$p.value

### ACHOLEPLASMATACEAE ###
wilcox.achol <- filter(wilcox.family, Family == "Acholeplasmataceae")

wilcox.achol.baseline <- filter(wilcox.achol, Group == "Baseline")
wilcox.achol.3wk <- filter(wilcox.achol, Group == "3 Wk Stress")

wilcox.test.achol <- wilcox.test(wilcox.achol.baseline$Abundance.Group, 
                                 wilcox.achol.3wk$Abundance.Group)$p.value

### AKKERMANSIACEAE ###
wilcox.akker <- filter(wilcox.family, Family == "Akkermansiaceae")

wilcox.akker.baseline <- filter(wilcox.akker, Group == "Baseline")
wilcox.akker.3wk <- filter(wilcox.akker, Group == "3 Wk Stress")

wilcox.test.akker <- wilcox.test(wilcox.akker.baseline$Abundance.Group, 
                                 wilcox.akker.3wk$Abundance.Group)$p.value

### CLOSTRIDIACEAE ###
wilcox.clos <- filter(wilcox.family, Family == "Clostridiaceae")

wilcox.clos.baseline <- filter(wilcox.clos, Group == "Baseline")
wilcox.clos.3wk <- filter(wilcox.clos, Group == "3 Wk Stress")

wilcox.test.clos <- wilcox.test(wilcox.clos.baseline$Abundance.Group, 
                                wilcox.clos.3wk$Abundance.Group)$p.value

### ERYSIPELOTRICHACEAE ###
wilcox.erys <- filter(wilcox.family, Family == "Erysipelotrichaceae")

wilcox.erys.baseline <- filter(wilcox.erys, Group == "Baseline")
wilcox.erys.3wk <- filter(wilcox.erys, Group == "3 Wk Stress")

wilcox.test.erys <- wilcox.test(wilcox.erys.baseline$Abundance.Group, 
                                wilcox.erys.3wk$Abundance.Group)$p.value

### LACHNOSPIRACEAE ###
wilcox.lachno <- filter(wilcox.family, Family == "Lachnospiraceae")

wilcox.lachno.baseline <- filter(wilcox.lachno, Group == "Baseline")
wilcox.lachno.3wk <- filter(wilcox.lachno, Group == "3 Wk Stress")

wilcox.test.lachno <- wilcox.test(wilcox.lachno.baseline$Abundance.Group, 
                                wilcox.lachno.3wk$Abundance.Group)$p.value

### LACTOBACILLACEAE ###
wilcox.lacto <- filter(wilcox.family, Family == "Lactobacillaceae")

wilcox.lacto.baseline <- filter(wilcox.lacto, Group == "Baseline")
wilcox.lacto.3wk <- filter(wilcox.lacto, Group == "3 Wk Stress")

wilcox.test.lacto <- wilcox.test(wilcox.lacto.baseline$Abundance.Group, 
                                 wilcox.lacto.3wk$Abundance.Group)$p.value

### MURIBACULACEAE ###
wilcox.muri <- filter(wilcox.family, Family == "Muribaculaceae")

wilcox.muri.baseline <- filter(wilcox.muri, Group == "Baseline")
wilcox.muri.3wk <- filter(wilcox.muri, Group == "3 Wk Stress")

wilcox.test.muri <- wilcox.test(wilcox.muri.baseline$Abundance.Group, 
                                 wilcox.muri.3wk$Abundance.Group)$p.value

### OSCILLOSPIRACEAE ###
wilcox.oscil <- filter(wilcox.family, Family == "Oscillospiraceae")

wilcox.oscil.baseline <- filter(wilcox.oscil, Group == "Baseline")
wilcox.oscil.3wk <- filter(wilcox.oscil, Group == "3 Wk Stress")

wilcox.test.oscil <- wilcox.test(wilcox.oscil.baseline$Abundance.Group, 
                                wilcox.oscil.3wk$Abundance.Group)$p.value

### PEPTOSCTREPTOCOCCACEAE ###
wilcox.pepto <- filter(wilcox.family, Family == "Peptostreptococcaceae")

wilcox.pepto.baseline <- filter(wilcox.pepto, Group == "Baseline")
wilcox.pepto.3wk <- filter(wilcox.pepto, Group == "3 Wk Stress")

wilcox.test.pepto <- wilcox.test(wilcox.pepto.baseline$Abundance.Group, 
                                 wilcox.pepto.3wk$Abundance.Group)$p.value


### RUMINOCOCCACEAE ###
wilcox.rumin <- filter(wilcox.family, Family == "Ruminococcaceae")

wilcox.rumin.baseline <- filter(wilcox.rumin, Group == "Baseline")
wilcox.rumin.3wk <- filter(wilcox.rumin, Group == "3 Wk Stress")

wilcox.test.rumin <- wilcox.test(wilcox.rumin.baseline$Abundance.Group, 
                                 wilcox.rumin.3wk$Abundance.Group)$p.value


Wilcox.Families <- c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", 
                     "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", 
                     "Muribaculaceae", "Oscillospiraceae", "Peptostreptococcaceae", "Ruminococcaceae")

Wilcox.pvalues <- c(wilcox.test.eubac, wilcox.test.achol, wilcox.test.akker, wilcox.test.clos, wilcox.test.erys,
                    wilcox.test.lachno, wilcox.test.lacto, wilcox.test.muri, wilcox.test.oscil, wilcox.test.pepto, 
                    wilcox.test.rumin)

Wilcox.pvalues.Bonferonni <- p.adjust(Wilcox.pvalues, method = "bonferroni", n = 13)

Wilcox.Baseline.3wk <- cbind(Wilcox.Families, Wilcox.pvalues)
Wilcox.Baseline.3wk <- cbind(Wilcox.Baseline.3wk, Wilcox.pvalues.Bonferonni)
Wilcox.Baseline.3wk <- as.data.frame(Wilcox.Baseline.3wk)
Wilcox.Baseline.3wk <- plyr::rename(Wilcox.Baseline.3wk, replace = c("Wilcox.Families" = "Family",
                                                                     "Wilcox.pvalues" = "Raw p value",
                                                                     "Wilcox.pvalues.Bonferonni" = "Bonferroni-adjusted p value"))


#### FAMILY LEVEL DIFFERENCES: STRESS ALONE VS STRESS + MUCIN ########################################
# Now I'll repeat this process, comparing the 4 week samples. I want to see if there are any major
# differences in families between the Control group (Stress Alone) and the Mucin-supplemented group
# (Stress + Mucin). I'll filter to include only these samples.

family.table.mucin.abundant.mucin <- filter(family.table.mucin.abundant, Group == "Stress Alone" | Group == "Stress + Mucin")
family.table.mucin.abundant.mucin$Group <- factor(family.table.mucin.abundant.mucin$Group, levels = c("Stress Alone", "Stress + Mucin"))

family.table.mucin.abundant.mucin <- filter(family.table.mucin.abundant.mucin, Family != "Peptostreptococcaceae")
family.table.mucin.abundant.mucin$Family <- factor(family.table.mucin.abundant.mucin$Family, levels = c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", 
                                                                                                        "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", 
                                                                                                        "Marinifilaceae", "Muribaculaceae", "Oscillospiraceae", "Rikenellaceae", 
                                                                                                        "Ruminococcaceae"))
```

```{r echo = FALSE}
knitr::kable(Wilcox.Baseline.3wk)
```

Statistically significant differences were observed in the following families: *Eubacterium*, *Acholeplasmataceae*, *Clostridiaceae*, *Lactobacillaceae*, *Muribaculaceae*, *Peptostreptococcaceae*, and *Ruminococcaceae*. Of these Families, *Muribaculaceae* and *Lactobacillaceae* make up the largest proportion of the microbiota. Of note, *Lactobacillaceae* was lower in the stressed group relative to baseline, which is consistent with previous experiments.
<br>

#### Stress Alone vs. Stress + Mucin
```{r include = FALSE}
#### FAMILY LEVEL DIFFERENCES: STRESS ALONE VS STRESS + MUCIN ########################################
# Now I'll repeat this process, comparing the 4 week samples. I want to see if there are any major
# differences in families between the Control group (Stress Alone) and the Mucin-supplemented group
# (Stress + Mucin). I'll filter to include only these samples.

family.table.mucin.abundant.mucin <- filter(family.table.mucin.abundant, Group == "Stress Alone" | Group == "Stress + Mucin")
family.table.mucin.abundant.mucin$Group <- factor(family.table.mucin.abundant.mucin$Group, levels = c("Stress Alone", "Stress + Mucin"))

family.table.mucin.abundant.mucin <- filter(family.table.mucin.abundant.mucin, Family != "Peptostreptococcaceae")
family.table.mucin.abundant.mucin$Family <- factor(family.table.mucin.abundant.mucin$Family, levels = c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", 
                                                                                                        "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", 
                                                                                                        "Marinifilaceae", "Muribaculaceae", "Oscillospiraceae", "Rikenellaceae", 
                                                                                                        "Ruminococcaceae"))

```

```{r echo = FALSE}
ggplot(family.table.mucin.abundant.mucin, aes(x = Abundance, y = Family, color = Group)) +
  geom_boxplot() +
  theme_bw() +
  scale_x_continuous(labels = scales::percent) +
  scale_y_discrete(limits = rev) +
  scale_color_manual(name = "Group", 
                     values = c("#521b92", "#4d8e00"), 
                     breaks = c("Stress Alone", "Stress + Mucin"), 
                     labels = c("Stress Alone", "Stress + Mucin")) +
  labs(x = "Relative Abundance", y = NULL) 
```


```{r include = FALSE}
### SIGNIFICANT DIFFERENCES: STRESS ALONE VS. STRESS + MUCIN ###
# I'll now use a Wilcoxon rank test to look for significant differences between each of these Family 
# groups. First, I'll organize the data so that abundance per Family is summarized on a per sample 
# basis. I'll also get Sample and Group metadata to add to summarized tables.
wilcox.family<- family.table.mucin.abundant.mucin %>%
  group_by(Family, Sample) %>%
  summarize(Abundance.Group = (sum(Abundance)))

Sample.Group <- select(family.table.mucin.abundant.mucin, Sample, Group)
Sample.Group <- Sample.Group %>%
  group_by(Sample) %>%
  slice(1L)

wilcox.family <- full_join(wilcox.family, Sample.Group, by = "Sample")

# Now I'll calculate Wilcoxon Ranked Sum p values for each Family.
table(wilcox.eubac$Group)
### EUBACTERIA ###
wilcox.eubac <- filter(wilcox.family, Family == "[Eubacterium] coprostanoligenes group")

wilcox.eubac.control <- filter(wilcox.eubac, Group == "Stress Alone")
wilcox.eubac.mucin <- filter(wilcox.eubac, Group == "Stress + Mucin")

wilcox.test.eubac <- wilcox.test(wilcox.eubac.control$Abundance.Group, 
                                 wilcox.eubac.mucin$Abundance.Group)$p.value

### ACHOLEPLASMATACEAE ###
wilcox.achol <- filter(wilcox.family, Family == "Acholeplasmataceae")

wilcox.achol.control <- filter(wilcox.achol, Group == "Stress Alone")
wilcox.achol.mucin <- filter(wilcox.achol, Group == "Stress + Mucin")

wilcox.test.achol <- wilcox.test(wilcox.achol.control$Abundance.Group, 
                                 wilcox.achol.mucin$Abundance.Group)$p.value

### AKKERMANSIACEAE ###
wilcox.akker <- filter(wilcox.family, Family == "Akkermansiaceae")

wilcox.akker.control <- filter(wilcox.akker, Group == "Stress Alone")
wilcox.akker.mucin <- filter(wilcox.akker, Group == "Stress + Mucin")

wilcox.test.akker <- wilcox.test(wilcox.akker.control$Abundance.Group, 
                                 wilcox.akker.mucin$Abundance.Group)$p.value

### CLOSTRIDIACEAE ###
wilcox.clos <- filter(wilcox.family, Family == "Clostridiaceae")

wilcox.clos.control <- filter(wilcox.clos, Group == "Stress Alone")
wilcox.clos.mucin <- filter(wilcox.clos, Group == "Stress + Mucin")

wilcox.test.clos <- wilcox.test(wilcox.clos.control$Abundance.Group, 
                                wilcox.clos.mucin$Abundance.Group)$p.value

### ERYSIPELOTRICHACEAE ###
wilcox.erys <- filter(wilcox.family, Family == "Erysipelotrichaceae")

wilcox.erys.control <- filter(wilcox.erys, Group == "Stress Alone")
wilcox.erys.mucin <- filter(wilcox.erys, Group == "Stress + Mucin")

wilcox.test.erys <- wilcox.test(wilcox.erys.control$Abundance.Group, 
                                wilcox.erys.mucin$Abundance.Group)$p.value

### LACHNOSPIRACEAE ###
wilcox.lachno <- filter(wilcox.family, Family == "Lachnospiraceae")

wilcox.lachno.control <- filter(wilcox.lachno, Group == "Stress Alone")
wilcox.lachno.mucin <- filter(wilcox.lachno, Group == "Stress + Mucin")

wilcox.test.lachno <- wilcox.test(wilcox.lachno.control$Abundance.Group, 
                                  wilcox.lachno.mucin$Abundance.Group)$p.value

### LACTOBACILLACEAE ###
wilcox.lacto <- filter(wilcox.family, Family == "Lactobacillaceae")

wilcox.lacto.control <- filter(wilcox.lacto, Group == "Stress Alone")
wilcox.lacto.mucin <- filter(wilcox.lacto, Group == "Stress + Mucin")

wilcox.test.lacto <- wilcox.test(wilcox.lacto.control$Abundance.Group, 
                                 wilcox.lacto.mucin$Abundance.Group)$p.value

### MARINIFILACEAE ###
wilcox.marin <- filter(wilcox.family, Family == "Marinifilaceae")

wilcox.marin.control <- filter(wilcox.marin, Group == "Stress Alone")
wilcox.marin.mucin <- filter(wilcox.marin, Group == "Stress + Mucin")

wilcox.test.marin <- wilcox.test(wilcox.marin.control$Abundance.Group, 
                                 wilcox.marin.mucin$Abundance.Group)$p.value

### MURIBACULACEAE ###
wilcox.muri <- filter(wilcox.family, Family == "Muribaculaceae")

wilcox.muri.control <- filter(wilcox.muri, Group == "Stress Alone")
wilcox.muri.mucin <- filter(wilcox.muri, Group == "Stress + Mucin")

wilcox.test.muri <- wilcox.test(wilcox.muri.control$Abundance.Group, 
                                wilcox.muri.mucin$Abundance.Group)$p.value

### OSCILLOSPIRACEAE ###
wilcox.oscil <- filter(wilcox.family, Family == "Oscillospiraceae")

wilcox.oscil.control <- filter(wilcox.oscil, Group == "Stress Alone")
wilcox.oscil.mucin <- filter(wilcox.oscil, Group == "Stress + Mucin")

wilcox.test.oscil <- wilcox.test(wilcox.oscil.control$Abundance.Group, 
                                 wilcox.oscil.mucin$Abundance.Group)$p.value

### RIKENELLACEAE ###
wilcox.riken <- filter(wilcox.family, Family == "Rikenellaceae")

wilcox.riken.control <- filter(wilcox.riken, Group == "Stress Alone")
wilcox.riken.mucin <- filter(wilcox.riken, Group == "Stress + Mucin")

wilcox.test.riken <- wilcox.test(wilcox.riken.control$Abundance.Group, 
                                 wilcox.riken.mucin$Abundance.Group)$p.value

### RUMINOCOCCACEAE ###
wilcox.rumin <- filter(wilcox.family, Family == "Ruminococcaceae")

wilcox.rumin.control <- filter(wilcox.rumin, Group == "Stress Alone")
wilcox.rumin.mucin <- filter(wilcox.rumin, Group == "Stress + Mucin")

wilcox.test.rumin <- wilcox.test(wilcox.rumin.control$Abundance.Group, 
                                 wilcox.rumin.mucin$Abundance.Group)$p.value


Wilcox.Families <- c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", 
                     "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", 
                     "Marinifilaceae", "Muribaculaceae", "Oscillospiraceae", "Rikenellaceae", 
                     "Ruminococcaceae")

Wilcox.pvalues <- c(wilcox.test.eubac, wilcox.test.achol, wilcox.test.akker, wilcox.test.clos, wilcox.test.erys,
                    wilcox.test.lachno, wilcox.test.lacto, wilcox.test.marin, wilcox.test.muri, wilcox.test.oscil,
                    wilcox.test.riken, wilcox.test.rumin)

Wilcox.pvalues.Bonferonni <- p.adjust(Wilcox.pvalues, method = "bonferroni", n = 12)

Wilcox.Control.Mucin <- cbind(Wilcox.Families, Wilcox.pvalues)
Wilcox.Control.Mucin <- cbind(Wilcox.Control.Mucin, Wilcox.pvalues.Bonferonni)
Wilcox.Control.Mucin <- as.data.frame(Wilcox.Control.Mucin)
Wilcox.Control.Mucin <- plyr::rename(Wilcox.Control.Mucin, replace = c("Wilcox.Families" = "Family",
                                                                     "Wilcox.pvalues" = "Unadjusted p value",
                                                                     "Wilcox.pvalues.Bonferonni" = "Bonferroni-adjusted p value"))
```

```{r echo = FALSE}
knitr::kable(Wilcox.Control.Mucin)
```

The *Rikenellaceae* and *Marinifilaceae* families were significantly different between Stress Alone and Stress + Mucin groups after correction for multiple comparisons. *Clostridiaceae* and *Lachnospiraceae* were different before adjustment for multiple comparisons. Of note, *Lactobacillaceae* was not significantly different between groups, indicating that Mucin supplementation is acting in a *Lactobacillaceae*-independent manner.

<br>

### Selection of ASVs that Distinguish Between Groups
I've broadly looked at differences in diversity and community composition with the Mucin Supplementation experiment for the Gautier TUMI Pilot. I'll now move on to identify ASVs that best discriminate between groups. There are two comparisons we are interested in for this experiment:

1. Baseline (n=24) vs. 3 Wk Stress samples (n=24)
2. Stress only (n=12) vs. Stress + Mucin (n=11)

ASVs will be selected by generating Random Forest models for each comparison. Models will be generated using training set data (a subset of the data containing 70% of samples), then model accuracy will be measured using test set data (the 30% of samples held out during model generation).
<br>

#### Baseline vs. 3 Week Stress Samples
```{r include = FALSE}
ps.mucin.stress.ASV <- subset_samples(ps.mucin, Group == "Baseline" | Group == "3 Wk Stress")
ASV.table.stress <- psmelt(ps.mucin.stress.ASV)

length(unique(ASV.table.stress$OTU)) #1795 unique ASVs across all samples.

# I want to limit the data set to only ASVs that are present in at least one group. To do this, I'll
# first summarize the Abundance of each ASV in each group, then select all ASVs that are >0 in any
# group.

ASV.table.summary.stress <- ASV.table.stress %>%
  group_by(OTU, Group) %>%
  summarize(Abundance.per.Group = (sum(Abundance))) %>%
  filter(any(Abundance.per.Group > 0)) # Removes any ASVs with an abundance of 0 in both groups.

length(unique(ASV.table.summary.stress$OTU)) # There are a total of 516 unique ASVs in these samples.


# Now I'll collect the ASV names for these ASVs. These will be used to merge only the ASV Abundance
# values for ASVs present in at least one group.
unique.ASVs.stress <- as.data.frame(unique(ASV.table.summary.stress$OTU))
unique.ASVs.stress <- plyr::rename(unique.ASVs.stress, replace = c("unique(ASV.table.summary.stress$OTU)" = "OTU"))

ASV.table.samples.stress <- left_join(unique.ASVs.stress, ASV.table.stress, by = "OTU")

length(unique(ASV.table.samples.stress$OTU)) # 516 unique ASVs as expected.


# I'll reshape the data into a wide format for analysis.
ASV.table.wide.stress <- select(ASV.table.samples.stress, OTU, Sample, Abundance, Group, Animal.Number)
ASV.table.wide.stress <- spread(ASV.table.wide.stress, key = OTU, value = Abundance)

# Now I'll split the data into Baseline and 3 Week Stress groups
ASV.table.wide.Baseline <- filter(ASV.table.wide.stress, Group == "Baseline")
ASV.table.wide.Baseline$Group <- factor(ASV.table.wide.Baseline$Group, levels = c("Baseline"))

ASV.table.wide.3Wk.Stress <- filter(ASV.table.wide.stress, Group == "3 Wk Stress")
ASV.table.wide.3Wk.Stress$Group <- factor(ASV.table.wide.3Wk.Stress$Group, levels = c("3 Wk Stress"))


### TRAINING AND TEST SET GENERATION: STRESS TREATMENT ###############################################
# There are 24 mice per group, which is a pretty good sample size for mouse experiments. Because of 
# this, I'll separate the data into training and test sets to better optimize the random forest 
# model and check model performance in an outside population.


# Set seed for reproducibility
#sample(1:1000, 1) # It selected 572
set.seed(572) 

# Now I'll separate each group into training and test sets.

### BASELINE ###
testIndex.Baseline <- createDataPartition(ASV.table.wide.Baseline$Group,
                                          p = 0.30,
                                          list = FALSE,
                                          times = 1)

featureTest.Baseline <- ASV.table.wide.Baseline[testIndex.Baseline,]
featureTrain.Baseline  <- ASV.table.wide.Baseline[-testIndex.Baseline,]


### 3 WEEK STRESS ###
testIndex.3Wk.Stress <- createDataPartition(ASV.table.wide.3Wk.Stress$Group,
                                            p = 0.30,
                                            list = FALSE,
                                            times = 1)

featureTest.3Wk.Stress <- ASV.table.wide.3Wk.Stress[testIndex.3Wk.Stress,]
featureTrain.3Wk.Stress  <- ASV.table.wide.3Wk.Stress[-testIndex.3Wk.Stress,]



# Finally, I'll combine each training and test partition together to make a complete training and 
# test set.

training.set.stress <- rbind(featureTrain.Baseline, featureTrain.3Wk.Stress) # Combine training sets together
test.set.stress <- rbind(featureTest.Baseline, featureTest.3Wk.Stress) # Combine test sets together


# Predictors
predictors.stress.Train <- select(training.set.stress, -Sample, -Group, -Animal.Number) # Remove metadata
predictors.stress.Test <- select(test.set.stress, -Sample, -Group, -Animal.Number) # Remove metadata

# Outcomes
outcome.stress.Train <- as.factor(training.set.stress$Group)
outcome.stress.Test <- as.factor(test.set.stress$Group)




#### CONSTRUCTING THE RANDOM FOREST MODEL ###########################################################
# The parameter I'm going to tune for the random forest model is mtry, the number of features sampled 
# at each node of the decision tree. For this random forest, I'll use ntree (the number of trees in 
# the forest) = 1000, which is a pretty large number but not too computationally taxing. Then I'll 
# look at a variety of different mtry values to see which minimizes the Out-of-Bag error rate.

# By default mtry is set to the square root of the number of features. There are 516 features in 
# this data set, meaning mtry be default is set to 23. I'll test mtry values around 23 to see if 
# this improves model error.


### TUNING MTRY ###
model.rf.mtry19 <- randomForest(x = predictors.stress.Train, y = outcome.stress.Train, ntree = 1000, mtry = 19)
model.rf.mtry19 # The OOB error rate is 0%

model.rf.mtry21 <- randomForest(x = predictors.stress.Train, y = outcome.stress.Train, ntree = 1000, mtry = 21)
model.rf.mtry21 # The OOB error rate is 0%

model.rf.mtry23 <- randomForest(x = predictors.stress.Train, y = outcome.stress.Train, ntree = 1000, mtry = 23)
model.rf.mtry23 # The OOB error rate is 0%

model.rf.mtry25 <- randomForest(x = predictors.stress.Train, y = outcome.stress.Train, ntree = 1000, mtry = 25)
model.rf.mtry25 # The OOB error rate is 0%

model.rf.mtry27 <- randomForest(x = predictors.stress.Train, y = outcome.stress.Train, ntree = 1000, mtry = 27)
model.rf.mtry27 # The OOB error rate is 0%

# All random forest models performed perfectly at correctly classifying Baseline and 3 week stress 
# samples, regardless of mtry value. Because of this, I'll just go with the standard mtry value
# of 23.

model.rf.stress.final <- randomForest(x = predictors.stress.Train, y = outcome.stress.Train, ntree = 1000, mtry = 23)
model.rf.stress.final




#### TESTING THE RANDOM FOREST MODEL ################################################################
# I now want to test the model on the test set data to see how well it can predict a data set it 
# hasn't seen before. 


pred <- predict(model.rf.stress.final, newdata = predictors.stress.Test)

prediction.stress.test <- select(test.set.stress, Sample, Group)
prediction.stress.test <- cbind(prediction.stress.test, pred)
prediction.stress.test <- plyr::rename(prediction.stress.test, replace = c("Group" = "Actual", 
                                                                           "pred" = "Predicted"))

View(prediction.stress.test)

# Overall, 8/8 (100%) of Baseline samples were correctly classified, while 7/8 (87.5%)  of 3 week
# stress samples were correctly classified. This suggests that the model is not really overfit
# and performs well on data it has not been trained on.

# I'll now generate ROC curves for both training and test sets to summarize model performance.

### TEST SET ###
pred.prob.stress <- predict(model.rf.stress.final, newdata = predictors.stress.Test, type = "prob") # Re-run prediction 
# looking at probabilities instead of classification.

pred.stress.data.frame <- as.data.frame(pred.prob.stress)
```

```{r echo = FALSE}
### TRAINING SET ###
roc(outcome.stress.Train, model.rf.stress.final$votes[,1], plot = TRUE, legacy.axes = TRUE, percent = TRUE, main = "Training Set ROC Curve",
    xlab = "False Positive Percentage", ylab = "True Positive Percentage", col = "#377eb8", lwd = 4, print.auc = TRUE)

# The RF model performs perfectly on the training set, with a 100% AUC. This is expected from
# the performance of the model on the training set data.


roc(outcome.stress.Test, pred.stress.data.frame$`3 Wk Stress`, plot = TRUE, legacy.axes = TRUE, percent = TRUE, main = "Test Set ROC Curve",
    xlab = "False Positive Percentage", ylab = "True Positive Percentage", col = "#377eb8", lwd = 4, print.auc = TRUE)
```
<br>

Performance of the random forest model was excellent with both training data (which was used to build the model) and test data (which was held out during model construction to later test the model for overfitting). The model correctly identified all training set data. The model also identified 8/8 Baseline samples and 7/8 3 Week Stress samples in the test set data. I'll now look at which ASVs best discriminate between groups.

``` {r include = FALSE}
#### IMPORTANCE VALUES FROM RANDOM FOREST MODEL ####################################################
# Now that I've generated the model, I'll look at which features best discriminate between groups as
# selected by the model. I'll do this by checking the importance values from the Random Forest model, 
# using Gini index as the measure of importance.

importance.rf.stress <- importance(model.rf.stress.final, type = 2)

# Format the importance data as a data frame
importance.rf.stress <- cbind(OTU = rownames(importance.rf.stress), importance.rf.stress)
rownames(importance.rf.stress) <- NULL
importance.rf.stress <- as.data.frame(importance.rf.stress)
importance.rf.stress$MeanDecreaseGini <- as.numeric(importance.rf.stress$MeanDecreaseGini)
str(importance.rf.stress$MeanDecreaseGini)

# I now want to combine the RF ranking data with the taxonomic information for identified ASVs. I'll
# pull this information from the phyloseq object metadata.

ASV.taxonomy <- as.data.frame(ps.mucin@tax_table)
ASV.taxonomy <- cbind(OTU = rownames(ASV.taxonomy), ASV.taxonomy)
rownames(ASV.taxonomy) <- NULL

importance.rf.stress.taxonomy <- left_join(importance.rf.stress, ASV.taxonomy, by = "OTU")

# This is the table of ASVs listed by mean decrease in Gini Index. I'll rank the features according
# to Gini index to make the data easier to interpret.

RF.ranking.stress <- importance.rf.stress.taxonomy %>%
  arrange(desc(MeanDecreaseGini))

RF.ranking.stress <- plyr::rename(RF.ranking.stress, replace = c("MeanDecreaseGini" = "Stress.Mean.Decrease.Gini"))
RF.ranking.stress$Stress.RF.Rank <- rank(-RF.ranking.stress$Stress.Mean.Decrease.Gini)

# I also want to add whether ASVs are enriched at baseline or after 3 weeks of stress treatment.
ASV.table.enrichment.stress <- ASV.table.stress %>%
  group_by(OTU, Group) %>%
  summarize(Average.Abundance.per.Group = (mean(Abundance))) %>%
  filter(any(Average.Abundance.per.Group > 0)) # Removes any ASVs with an abundance of 0 in both groups.

ASV.enrichment.baseline <- filter(ASV.table.enrichment.stress, Group == "Baseline")
ASV.enrichment.baseline <- plyr::rename(ASV.enrichment.baseline, replace = c("Average.Abundance.per.Group" = "Baseline.Average.Abundance.per.Group"))
ASV.enrichment.baseline <- select(ASV.enrichment.baseline, OTU, Baseline.Average.Abundance.per.Group)

ASV.enrichment.3Wk.stress <- filter(ASV.table.enrichment.stress, Group == "3 Wk Stress")
ASV.enrichment.3Wk.stress <- plyr::rename(ASV.enrichment.3Wk.stress, replace = c("Average.Abundance.per.Group" = "3Wk.stress.Average.Abundance.per.Group"))
ASV.enrichment.3Wk.stress <- select(ASV.enrichment.3Wk.stress, OTU, '3Wk.stress.Average.Abundance.per.Group')

ASV.enrichment.stress <- full_join(ASV.enrichment.baseline, ASV.enrichment.3Wk.stress, by = "OTU")
ASV.enrichment.stress$Enrichment.Stress <- NA
ASV.enrichment.stress$Enrichment.Stress[ASV.enrichment.stress$Baseline.Average.Abundance.per.Group > ASV.enrichment.stress$`3Wk.stress.Average.Abundance.per.Group`] <- "Baseline"
ASV.enrichment.stress$Enrichment.Stress[ASV.enrichment.stress$Baseline.Average.Abundance.per.Group < ASV.enrichment.stress$`3Wk.stress.Average.Abundance.per.Group`] <- "3 Wk Stress"

ASV.enrichment <- select(ASV.enrichment.stress, OTU, Enrichment.Stress)

RF.ranking.stress <- full_join(RF.ranking.stress, ASV.enrichment, by = "OTU")

RF.ranking.stress$OTU <- factor(RF.ranking.stress$OTU, levels = RF.ranking.stress$OTU)

# I'll make new names for each figure, which will concatenate the ASV number and the Family
# designation for each ASV.
RF.ranking.stress$ASV.Family <- NA
RF.ranking.stress$ASV.Family <- paste(RF.ranking.stress$OTU, " (", RF.ranking.stress$Family, ")")
RF.ranking.stress$ASV.Family <- factor(RF.ranking.stress$ASV.Family, levels = RF.ranking.stress$ASV.Family)


ggplot(RF.ranking.stress, aes(x = Stress.Mean.Decrease.Gini, y = ASV.Family, color = Enrichment.Stress)) +
  geom_point(size = 3) +
  labs(title = "Top Ranked Random Forest Predictors", x = "Mean Decrease in Node Impurity (Gini)", y = NULL) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"))

# To reduce the number of ASVs graphed on the figure, I'm going to set an arbitrarily cutoff 
# for the Gini Index to include only the most important ASVs.

RF.ranking.stress.figure <- filter(RF.ranking.stress, Stress.Mean.Decrease.Gini > 0.12)

```

```{r echo = FALSE}
ggplot(RF.ranking.stress.figure, aes(x = Stress.Mean.Decrease.Gini, y = ASV.Family, color = Enrichment.Stress)) +
  geom_point(size = 3) +
  labs(title = "Top Ranked Random Forest Predictors", x = "Mean Decrease in Node Impurity (Gini)", y = NULL, color = "Enrichment") +
  scale_color_manual(name = "Enrichment", 
                     values = c("#941650", "#1c7798"),
                     breaks = c("Baseline", "3 Wk Stress"), 
                     labels = c("Baseline", "3 Wk Stress")) +
  scale_y_discrete(limits=rev) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"))
```
<br>

Most ASVs selected by the model were from the *Lachnospiraceae* Family. The top 5 ASVs selected by the model seem to be more important to the importance of the model compared to other ASVs as measured by the Gini Index. This index measures the decrease in node impurity averaged across all trees in the forest, with a higher mean decrease in Gini indicating higher variable importance.

To confirm this, a separate random forest model was generated using only the top 5 ASVs selected by the original model. This new model was also able to correctly discriminate between Baseline and 3 Week Stress Groups (perfect on training set, 15/16 correct on test set), indicating that these ASVs are sufficient to discriminate between groups.
<br>

#### Stress Alone vs. Stress + Mucin 
```{r include = FALSE}
#### ORGANIZATION OF ASVs FROM CONTROL AND MUCIN-SUPPLEMENTED SAMPLES ################################
# I'll now move on to the second half of the experiment. Here I'll be classifying 4 week samples only
# as either Control or Mucin-supplemented groups.

ps.mucin.ASV <- subset_samples(ps.mucin, Group == "Stress Alone" | Group == "Stress + Mucin")
ASV.table.mucin <- psmelt(ps.mucin.ASV)

length(unique(ASV.table.stress$OTU)) #1795 unique ASVs across all samples.

# I want to limit the data set to only ASVs that are present in at least one group. To do this, I'll
# first summarize the Abundance of each ASV in each group, then select all ASVs that are >0 in any
# group.

ASV.table.summary.mucin <- ASV.table.mucin %>%
  group_by(OTU, Group) %>%
  summarize(Abundance.per.Group = (sum(Abundance))) %>%
  filter(any(Abundance.per.Group > 0)) # Removes any ASVs with an abundance of 0 in both groups.

length(unique(ASV.table.summary.mucin$OTU)) # There are a total of 498 unique ASVs in these samples.


# Now I'll collect the ASV names for these ASVs. These will be used to merge only the ASV Abundance
# values for ASVs present in at least one group.
unique.ASVs.mucin <- as.data.frame(unique(ASV.table.summary.mucin$OTU))
unique.ASVs.mucin <- plyr::rename(unique.ASVs.mucin, replace = c("unique(ASV.table.summary.mucin$OTU)" = "OTU"))

ASV.table.samples.mucin <- left_join(unique.ASVs.mucin, ASV.table.mucin, by = "OTU")

length(unique(ASV.table.samples.mucin$OTU)) # 498 unique ASVs as expected.


# I'll reshape the data into a wide format for analysis.
ASV.table.wide.mucin <- select(ASV.table.samples.mucin, OTU, Sample, Abundance, Group, Animal.Number)
ASV.table.wide.mucin <- spread(ASV.table.wide.mucin, key = OTU, value = Abundance)

# Now I'll split the data into Control and Mucin groups
ASV.table.wide.Control <- filter(ASV.table.wide.mucin, Group == "Stress Alone")
ASV.table.wide.Control$Group <- factor(ASV.table.wide.Control$Group, levels = c("Stress Alone"))

ASV.table.wide.Mucin <- filter(ASV.table.wide.mucin, Group == "Stress + Mucin")
ASV.table.wide.Mucin$Group <- factor(ASV.table.wide.Mucin$Group, levels = c("Stress + Mucin"))


### TRAINING AND TEST SET GENERATION: STRESS TREATMENT ###############################################
# There are 12 mice in the Control group and 11 mice in the Mucin-supplemented group. This is smaller
# than the sample size for the Baseline vs Stress comparison, but should still be ok for model
# building. I'll now separate the data into training and test sets to better optimize the random forest 
# model and check model performance in an outside population.


# Set seed for reproducibility
#sample(1:1000, 1) # It selected 389
set.seed(389) 

# Now I'll separate each group into training and test sets.

### CONTROL ###
testIndex.Control <- createDataPartition(ASV.table.wide.Control$Group,
                                          p = 0.30,
                                          list = FALSE,
                                          times = 1)

featureTest.Control <- ASV.table.wide.Control[testIndex.Control,]
featureTrain.Control  <- ASV.table.wide.Control[-testIndex.Control,]


### MUCIN ###
testIndex.Mucin <- createDataPartition(ASV.table.wide.Mucin$Group,
                                            p = 0.30,
                                            list = FALSE,
                                            times = 1)

featureTest.Mucin <- ASV.table.wide.Mucin[testIndex.Mucin,]
featureTrain.Mucin  <- ASV.table.wide.Mucin[-testIndex.Mucin,]



# Finally, I'll combine each training and test partition together to make a complete training and 
# test set.

training.set.mucin <- rbind(featureTrain.Control, featureTrain.Mucin) # Combine training sets together
test.set.mucin <- rbind(featureTest.Control, featureTest.Mucin) # Combine test sets together


# Predictors
predictors.mucin.Train <- select(training.set.mucin, -Sample, -Group, -Animal.Number) # Remove metadata
predictors.mucin.Test <- select(test.set.mucin, -Sample, -Group, -Animal.Number) # Remove metadata

# Outcomes
outcome.mucin.Train <- as.factor(training.set.mucin$Group)
outcome.mucin.Test <- as.factor(test.set.mucin$Group)




#### CONSTRUCTING THE RANDOM FOREST MODEL ###########################################################
# The parameter I'm going to tune for the random forest model is mtry, the number of features sampled 
# at each node of the decision tree. For this random forest, I'll use ntree (the number of trees in 
# the forest) = 1000, which is a pretty large number but not too computationally taxing. Then I'll 
# look at a variety of different mtry values to see which minimizes the Out-of-Bag error rate.

# By default mtry is set to the square root of the number of features. There are 498 features in 
# this data set, meaning mtry be default is set to 22. I'll test mtry values around 22 to see if 
# this improves model error.


### TUNING MTRY ###
model.rf.mtry18 <- randomForest(x = predictors.mucin.Train, y = outcome.mucin.Train, ntree = 1000, mtry = 18)
model.rf.mtry18 # The OOB error rate is 13.33%

model.rf.mtry20 <- randomForest(x = predictors.mucin.Train, y = outcome.mucin.Train, ntree = 1000, mtry = 20)
model.rf.mtry20 # The OOB error rate is 13.33%

model.rf.mtry22 <- randomForest(x = predictors.mucin.Train, y = outcome.mucin.Train, ntree = 1000, mtry = 22)
model.rf.mtry22 # The OOB error rate is 20%

model.rf.mtry24 <- randomForest(x = predictors.mucin.Train, y = outcome.mucin.Train, ntree = 1000, mtry = 24)
model.rf.mtry24 # The OOB error rate is 13.33%

model.rf.mtry26 <- randomForest(x = predictors.mucin.Train, y = outcome.mucin.Train, ntree = 1000, mtry = 26)
model.rf.mtry26 # The OOB error rate is 6.67%

# These models performed more poorly than the earlier models, but still performed well, with OOB 
# error rates between 7-20%. Repeating these models several times, it looks like there 
# heterogeneity in the error rate for each model, so they're all roughly similar. I'll go with 
# mtry == 26, because it was slightly more consistent with a lower OOB error rate.


model.rf.mucin.final <- randomForest(x = predictors.mucin.Train, y = outcome.mucin.Train, ntree = 1000, mtry = 26)
model.rf.mucin.final




#### TESTING THE RANDOM FOREST MODEL ################################################################
# I now want to test the model on the test set data to see how well it can predict a data set it 
# hasn't seen before. 

pred <- predict(model.rf.mucin.final, newdata = predictors.mucin.Test)

prediction.mucin.test <- select(test.set.mucin, Sample, Group)
prediction.mucin.test <- cbind(prediction.mucin.test, pred)
prediction.mucin.test <- plyr::rename(prediction.mucin.test, replace = c("Group" = "Actual", 
                                                                           "pred" = "Predicted"))

View(prediction.mucin.test)

# Overall, 4/4 (100%) of Control (Stress Alone) samples were correctly classified, while 4/4 (100%)  
# of Mucin-supplemented (Stress + Mucin) samples were correctly classified. This suggests that the 
# model is not really overfit and performs well on data it has not been trained on.

# I'll now generate ROC curves for both training and test sets to summarize model performance.


### TEST SET ###
pred.prob.mucin <- predict(model.rf.mucin.final, newdata = predictors.mucin.Test, type = "prob") # Re-run prediction 
# looking at probabilities instead of classification.

pred.mucin.data.frame <- as.data.frame(pred.prob.mucin)
```

```{r echo = FALSE}
### TRAINING SET ###
roc(outcome.mucin.Train, model.rf.mucin.final$votes[,1], plot = TRUE, legacy.axes = TRUE, percent = TRUE, main = "Training Set ROC Curve",
    xlab = "False Positive Percentage", ylab = "True Positive Percentage", col = "#377eb8", lwd = 4, print.auc = TRUE)

roc(outcome.mucin.Test, pred.mucin.data.frame$`Stress + Mucin`, plot = TRUE, legacy.axes = TRUE, percent = TRUE, main = "Test Set ROC Curve",
    xlab = "False Positive Percentage", ylab = "True Positive Percentage", col = "#377eb8", lwd = 4, print.auc = TRUE)

```
<br>

This random forest model did not perform quite as well on the training set data compared to the Baseline vs. 3 Wk Stress. While this new model had a larger out-of-bag error rate compared to that model, it was able to correctly classify all samples in the test data set, indicating that it still performs well. The slightly impaired performance may reflect the smaller sample size for this comparison.

I'll now look at the ASVs selected by this model.
<br>

```{r include = FALSE}
#### IMPORTANCE VALUES FROM RANDOM FOREST MODEL ####################################################
# Now that I've generated the model, I'll look at which features best discriminate between groups as
# selected by the model. I'll do this by checking the importance values from the Random Forest model, 
# using Gini index as the measure of importance.

importance.rf.mucin <- importance(model.rf.mucin.final, type = 2)

# Format the importance data as a data frame
importance.rf.mucin <- cbind(OTU = rownames(importance.rf.mucin), importance.rf.mucin)
rownames(importance.rf.mucin) <- NULL
importance.rf.mucin <- as.data.frame(importance.rf.mucin)
importance.rf.mucin$MeanDecreaseGini <- as.numeric(importance.rf.mucin$MeanDecreaseGini)
str(importance.rf.mucin$MeanDecreaseGini)

# I now want to combine the RF ranking data with the taxonomic information for identified ASVs. I'll
# pull this information from the phyloseq object metadata.

importance.rf.mucin.taxonomy <- left_join(importance.rf.mucin, ASV.taxonomy, by = "OTU")

# This is the table of ASVs listed by mean decrease in Gini Index. I'll rank the features according
# to Gini index to make the data easier to interpret.

RF.ranking.mucin <- importance.rf.mucin.taxonomy %>%
  arrange(desc(MeanDecreaseGini))

RF.ranking.mucin <- plyr::rename(RF.ranking.mucin, replace = c("MeanDecreaseGini" = "Mucin.Mean.Decrease.Gini"))
RF.ranking.mucin$Mucin.RF.Rank <- rank(-RF.ranking.mucin$Mucin.Mean.Decrease.Gini)


# I also want to add whether ASVs are enriched in the Control group or in the Mucin-supplemented group.
ASV.table.enrichment.mucin <- ASV.table.mucin %>%
  group_by(OTU, Group) %>%
  summarize(Average.Abundance.per.Group = (mean(Abundance))) %>%
  filter(any(Average.Abundance.per.Group > 0)) # Removes any ASVs with an abundance of 0 in both groups.

ASV.enrichment.Control <- filter(ASV.table.enrichment.mucin, Group == "Stress Alone")
ASV.enrichment.Control <- plyr::rename(ASV.enrichment.Control, replace = c("Average.Abundance.per.Group" = "Control.Average.Abundance.per.Group"))
ASV.enrichment.Control <- select(ASV.enrichment.Control, OTU, Control.Average.Abundance.per.Group)

ASV.enrichment.Mucin <- filter(ASV.table.enrichment.mucin, Group == "Stress + Mucin")
ASV.enrichment.Mucin <- plyr::rename(ASV.enrichment.Mucin, replace = c("Average.Abundance.per.Group" = "Mucin.Average.Abundance.per.Group"))
ASV.enrichment.Mucin <- select(ASV.enrichment.Mucin, OTU, 'Mucin.Average.Abundance.per.Group')

ASV.enrichment.mucin <- full_join(ASV.enrichment.Control, ASV.enrichment.Mucin, by = "OTU")
ASV.enrichment.mucin$Enrichment.Mucin <- NA
ASV.enrichment.mucin$Enrichment.Mucin[ASV.enrichment.mucin$Control.Average.Abundance.per.Group > ASV.enrichment.mucin$Mucin.Average.Abundance.per.Group] <- "Stress Alone"
ASV.enrichment.mucin$Enrichment.Mucin[ASV.enrichment.mucin$Control.Average.Abundance.per.Group < ASV.enrichment.mucin$Mucin.Average.Abundance.per.Group] <- "Stress + Mucin"

ASV.enrichment <- select(ASV.enrichment.mucin, OTU, Enrichment.Mucin)

RF.ranking.mucin <- full_join(RF.ranking.mucin, ASV.enrichment, by = "OTU")

RF.ranking.mucin$OTU <- factor(RF.ranking.mucin$OTU, levels = RF.ranking.mucin$OTU)

# I'll make new names for each figure, which will concatenate the ASV number and the Family
# designation for each ASV.
RF.ranking.mucin$ASV.Family <- NA
RF.ranking.mucin$ASV.Family <- paste(RF.ranking.mucin$OTU, " (", RF.ranking.mucin$Family, ")")
RF.ranking.mucin$ASV.Family <- factor(RF.ranking.mucin$ASV.Family, levels = RF.ranking.mucin$ASV.Family)


ggplot(RF.ranking.mucin, aes(x = Mucin.Mean.Decrease.Gini, y = ASV.Family, color = Enrichment.Mucin)) +
  geom_point(size = 3) +
  labs(title = "Top Ranked Random Forest Predictors", x = "Mean Decrease in Node Impurity (Gini)", y = NULL) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"))


# To reduce the number of ASVs graphed on the figure, I'm going to set an arbitrarily cutoff 
# for the Gini Index to include only the most important ASVs.

RF.ranking.mucin.figure <- filter(RF.ranking.mucin, Mucin.Mean.Decrease.Gini > 0.07)

RF.ranking.mucin.figure$Enrichment.Mucin <- factor(RF.ranking.mucin.figure$Enrichment.Mucin, levels = c("Stress Alone", "Stress + Mucin"))

```

```{r echo = FALSE}
ggplot(RF.ranking.mucin.figure, aes(x = Mucin.Mean.Decrease.Gini, y = ASV.Family, color = Enrichment.Mucin)) +
  geom_point(size = 3) +
  labs(title = "Top Ranked Random Forest Predictors", x = "Mean Decrease in Node Impurity (Gini)", y = NULL, color = "Enrichment") +
  scale_color_manual(name = "Enrichment", 
                     values = c("#521b92", "#4d8e00"),
                     breaks = c("Stress Alone", "Stress + Mucin"), 
                     labels = c("Stress Alone", "Stress + Mucin")) +
  scale_y_discrete(limits=rev) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"))
```

The ASVs that discriminate between Stress Alone and Stress + Mucin groups are once again dominated by *Lachnospiraceae* Family members. I'll check to see how much overlap there is between the most important ASVs for the Baseline vs. 3 Wk Stress comparison and the Stress Alone vs. Stress + Mucin comparison.
<br>

```{r include = FALSE}

RF.ranking.combined <- select(RF.ranking.stress, OTU, Stress.Mean.Decrease.Gini, Stress.RF.Rank, Enrichment.Stress)

RF.ranking.combined <- full_join(RF.ranking.combined, RF.ranking.mucin, by = "OTU")
RF.ranking.combined <- select(RF.ranking.combined, OTU, Stress.Mean.Decrease.Gini, Mucin.Mean.Decrease.Gini,
                              Stress.RF.Rank, Mucin.RF.Rank, Enrichment.Stress, Enrichment.Mucin, everything() )                     

View(RF.ranking.combined)
# It doesn't look like there's a whole lot of overlap between data sets. To look for sure, I'll arbitrarily
# limit the data set to ASVs that were in the top 50 for each outcome by mean decrease in Gini Index.

RF.ranking.combined.important <- filter(RF.ranking.combined, Stress.RF.Rank < 50 & Mucin.RF.Rank < 50)                              

```

```{r echo = FALSE}
kable(RF.ranking.combined.important)
```

There isn't a whole lot of overlap between ASVs, and those ASVs that do overlap are not consistenly enriched in the same direction (i.e., enriched in both Baseline and Stress + Mucin or in both 3 Wk Stress and Stress Alone). This indicates that mucin supplementation doesn't seem to be restoring the microbiota to the same community composition seen at Baseline.

<br>

### Conclusions
This experiment focused on the role of stress and mucin supplementation on the diversity and community composition of the intestinal microbiota. Both stress and mucin supplementation significantly affected microbial beta diversity. Interestingly, changes in microbial evenness and beta diversity were strongest between 4 week samples and other timepoints, regardless of whether the samples were supplemented with mucin. This indicates that a mucin-independent feature (maybe diet) is driving this change.
<br>

Significant differences in the Family-level composition were observed between Baseline and 3 Week Stress samples, particularly in the *Lactobacillaceae* and *Muribaculaceae* Families. This difference in *Lactobacillaceae* is consistent with previous results from the lab. Random Forest modeling was able to clearly discriminate between Baseline and 3 Week Stress groups using the 16S data and identified several ASVs (mainly *Lachnospiraceae* and one *Lactobacillaceae*) which separate groups.
<br>

Signifcant differences were also observed at the Family level (*Marinifilaceae* and *Rikenellaceae*) with Stress Alone and Stress + Mucin groups. Some trends were observed with other Families, but they were not significant after correction with multiple comparisons. It's worth noting that the sample size is smaller in this analysis than that for Baseline vs. 3 Week Stress. Random forest modeling also identified ASVs associated with Mucin supplementation, but most of these ASVs, including *Lactobacillaceae* were not the same ASVs identified in the Baseline vs. 3 Week Stress comparison. This suggests that mucin supplementation may be acting by a different mechanism than that identified by previous studies in the lab.