---
title: Mucin Supplementation Analysis Report
author: "G. Brett Moreau"
date: "December 15, 2021"
output:
  html_document:
    toc: yes
  pdf_document:
    toc: yes
---
``` {r setup, include = FALSE}
#### PACKAGES ################################################
#if (!requireNamespace("BiocManager", quietly = TRUE))
#  install.packages("BiocManager")
#BiocManager::install()
packageVersion("BiocManager") # I'm using version 1.30.16

#BiocManager::install("phyloseq")
library(phyloseq)
packageVersion("phyloseq") # I'm using version 1.38.0

#install.packages("tidyverse")
library(tidyverse)
packageVersion("tidyverse") # I'm using version 1.3.1

#install.packages("ggplot2")
library(ggplot2)
packageVersion("ggplot2") # I'm using version 3.3.5

#BiocManager::install("microbiome")
library(microbiome) 
packageVersion("microbiome") # I'm using version 1.16.0

#install.packages("vegan")
library(vegan)
packageVersion("vegan") # I'm using version 2.5.7

#install.packages("caret")
library(caret)
packageVersion("caret") # I'm using version 6.0.90

#install.packages("randomForest")
library(randomForest)
packageVersion("randomForest") # I'm using version 4.6.14

#install.packages("pROC")
library(pROC)
packageVersion("pROC") # I'm using version 1.18.0
```
### Summary of Results

* There were no differences in richness between samples, but evenness was increased in 4 week samples (both Stress Alone and Stress + Mucin) compared to earlier samples.

* Differences in beta diversity were observed between all groups, particularly between samples collected at four weeks and samples collected at earlier timepoints.

* Phylum- and Family-level community composition changes were observed across groups. Several Families (including *Lactobacillaceae* and *Muribaculaceae*) were changed between Baseline and 3 week stress conditions, while a couple (*Rikenellaceae* and *Marinifilaceae*) were changed between Stress Alone and Stress + Mucin conditions.

* 
<br>

### Introduction
The purpose of this experiment is to investigate the microbiome of stressed mice whose diet has or has not been supplemented  with mucin. Behavioral experiments have observed some changes in stress behavior after mucin supplementation. Because of 

<br>

### Characterization of the Data Set
There are 71 samples in this experiment. Samples were divided into two groups (Control and Mucin-supplemented) and three different timepoints (pre-stress, 3 weeks post-stress, and 4 weeks post-stress). Four week samples are broken into two groups: Stress Alone, which is the control group, and Stress + Mucin, which is has a diet supplemented with 5% mucin.  There are 24 mice at Baseline and 3 Weeks Stress,  12 mice for Stress Alone, and 11 mice for Stress + Mucin (Sample #167 was removed due to low read counts), giving our 71 samples.
<br>

Below is a general overview of the total number of reads in the data set, the mean and median number of reads/sample, and the range of reads across this data set.
<br>

```{r include = FALSE}
load(file = "/Users/gbm5pn/Documents/GitHub/Gautier-TUMI-Pilot-16S/results/phyloseq objects by experiment.RData")

ps.mucin.prop <- transform_sample_counts(ps.mucin, function(ASV) ASV/sum(ASV)) 

total.reads <- sample_sums(ps.mucin)
```

```{r}
sum(total.reads)
mean(total.reads) 
median(total.reads)
range(total.reads) 
```

<br>

### Alpha Diversity 
I'll first investigate alpha diversity, which examines diversity within a given sample. Alpha diversity consists of two major measures:

* **Richness**: the number of taxonomic groups within a given sample.

* **Evenness**: how evenly abundances of taxonomic groups are distributed in a given sample.
<br>

The DADA2 pipeline removes singlets (sequences detected only once) and doublets (sequences detected twice) from the data set, as it does not have enough confidence to determine whether these sequences are real or artifacts. Because of this, measures of richness fail explicitly and are difficult to interpret. I'll focus mainly on measures of evenness but give measures of richness a glance to see if there are any major differences.
<br><br>

#### Richness
```{r echo = FALSE}
plot_richness(ps.mucin, x = "Group", measures = c("Observed"), 
              color = "Group") + 
  geom_point() +
  geom_boxplot() + 
  theme_bw() +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_blank()) +
  scale_color_discrete(name = "Group", 
                       breaks = c("Baseline", "3 Wk Stress", "Stress Alone", "Stress + Mucin"), 
                       labels = c("Baseline", "3 Wk Stress", "Stress Alone", "Stress + Mucin")) 


richness <- estimate_richness(ps.mucin)
pairwise.wilcox.test(richness$Observed, p.adjust.method = "bonferroni", sample_data(ps.mucin)$Group)

```
<br>

There are no major differences in alpha diversity between groups.
<br><br>

#### Evenness
```{r echo = FALSE}
# First I'll organize the data.
sample.data.mucin <- ps.mucin@sam_data
sample.data.mucin <- cbind("Sample.Names" = rownames(sample.data.mucin), sample.data.mucin)

evenness<- evenness(ps.mucin, index = c("pielou"))
evenness<- cbind("Sample.Names" = rownames(evenness), evenness)

sample.data.mucin <- full_join(sample.data.mucin, evenness, by = "Sample.Names")


### PIELOU ###
ggplot(sample.data.mucin, aes(x = Group, y = pielou, color = Group)) +
  geom_point() +
  geom_boxplot() +
  theme_bw() +
  labs(title = "Microbial Evenness Across Groups", x = NULL, y = "Pielou Eveness Index")
  theme(legend.position = "none")
```
<br>

Baseline and 3 Week Stress samples look similar to each other, while 4 week samples (Stress Alone and Stress + Mucin) appear to have higher evenness . A Pairwise Wilcoxon Rank Sum Test with Bonferroni correction from multiple comparisons was performed to test for differences between groups.
<br>

```{r echo = FALSE}
pairwise.wilcox.test(evenness$pielou, p.adjust.method = "bonferroni", sample_data(ps.mucin)$Group)
```
<br>

There was no significant difference in evenness between Baseline and 3 Week Stress samples. However, both 4 week samples (Stress Alone and Stress + Mucin) were significantly higher than either Baseline or 3 Week Stress groups. No significant difference in evenness was observed between Stress Alone and Stress + Mucin groups.

<br>

### Beta Diversity
I'll now investigate differences in beta diversity, or diversity between different samples in the experiment. I'll use Bray-Curtis Dissimilarity as my measure for beta diversity in this study, and I'll use Non-metric Multidimensional Scaling (NMDS) to visualize dissimilarity distances between samples.

``` {r include = FALSE}
#sample(1:1000, 1) # It selected 966
set.seed(966) # Set seed for reproducibility

# First, let's ordinate Bray-Curtis Dissimilarity using NMDS, then plot this ordination to visualize
# separation between different samples.

ord.nmds.bray <- ordinate(ps.mucin.prop, method = "NMDS", distance = "bray")
```

```{r echo = FALSE}
plot_ordination(ps.mucin.prop, ord.nmds.bray, color = "Group", 
                title = "Bray-Curtis Dissimilarity") +
  theme_bw() +
  scale_color_manual(name = "Group", 
                     values = c("#941650", "#1c7798", "#521b92", "#4d8e00"),
                       breaks = c("Baseline", "3 Wk Stress", "Stress Alone", "Stress + Mucin"), 
                       labels = c("Baseline", "3 Wk Stress", "Stress Alone", "Stress + Mucin")) +
  theme(aspect.ratio = 1, plot.title = element_text(hjust = 0.5)) 
```
<br>

Looking at the clustering, it appears that groups cluster distinctly from each other. Baseline and 3 Week Stress groups are relatively similar to each other, and 4 week samples (Stress Alone and Stress + Mucin) are relatively similar to each other. Baseline and 3 Week samples are quite distinct from 4 week samples.
<br>

I'll now use ADONIS to calculate statistical differences between groups.
<br>

```{r include = FALSE}
adonis(distance(ps.mucin.prop, method = "bray") ~sample_data(ps.mucin.prop)$Group)

# The PERMANOVA found significant differences in clustering (p = 0.001), but this doesn't specifiy
# which groups are significantly different from each other. To test this, I'll perform pairwise
# PERMANOVAs with multiple comparisons correction.

permanova.combinations <- combn(x = levels(ps.mucin.prop@sam_data$Group), m = 2)
permanova.p.bray <- vector(length = 6)
for(i in 1:ncol(permanova.combinations)){
  ps.subs.bray <- subset_samples(ps.mucin.prop, Group %in% permanova.combinations[,i])
  metadata.subs.bray <- as(sample_data(ps.subs.bray), "data.frame")
  pairwise.bray <- adonis(distance(ps.subs.bray, method = "bray") ~ Group, data = metadata.subs.bray)
  permanova.p.bray[i] <- pairwise.bray$aov.tab[1,ncol(pairwise.bray$aov.tab)]
}
combinations <- c("Baseline/3 Wk Stress", "Baseline/Stress Alone", "Baseline/Stress + Mucin", 
                  "3 Wk Stress/Stress Alone", "3 Wk Stress/Stress + Mucin",
                  "Stress Alone/Stress + Mucin")

permanova.FDR.bray <- p.adjust(permanova.p.bray, method = "bonferroni")
permanova.bray.table <- cbind(combinations, permanova.p.bray)
permanova.bray.table <- cbind(permanova.bray.table, permanova.FDR.bray)

```

```{r echo = FALSE}
library(knitr)
kable(permanova.bray.table)
```
<br>
The PERMANOVA data confirms the initial impressions from the NMDS plot: All groups cluster separately from each other and are statistically significant from every other group.

<br>

### Changes in Community Composition
There appear to be some broad differences in diversity between groups. I'll now see if this is reflected by any broad changes in community composition across groups. I'll start by looking at the phylum level, then move on to differences at the Family level. This analysis will focus on taxa that make up at least 1% of the relative abundance in at least one group.

```{r include = FALSE}
ps.mucin.prop.phylum <- tax_glom(ps.mucin.prop, "Phylum", NArm = FALSE)
phylum.table.mucin <- psmelt(ps.mucin.prop.phylum) # Organize in long format for ggplot.

# There are 24 samples for "Baseline" and "3 Wk Stress" groups, 12
# samples for the "Stress Alone" group, and 11 samples for the "Stress + Mucin" group. I'll 
# separate these groups to generate relative abundances as a percentage by dividing by the total
# number of samples in each group.

phylum.table.mucin.baseline.3wk <- filter(phylum.table.mucin, Group == "Baseline" | Group == "3 Wk Stress")
phylum.table.mucin.baseline.3wk$Relative.Abundance <- NA
phylum.table.mucin.baseline.3wk$Relative.Abundance <- phylum.table.mucin.baseline.3wk$Abundance / 24 *100


phylum.table.stress <- filter(phylum.table.mucin, Group == "Stress Alone")
phylum.table.stressRelative.Abundance <- NA
phylum.table.stress$Relative.Abundance <- phylum.table.stress$Abundance / 12 *100


phylum.table.stress.mucin <- filter(phylum.table.mucin, Group == "Stress + Mucin")
phylum.table.stress.mucinRelative.Abundance <- NA
phylum.table.stress.mucin$Relative.Abundance <- phylum.table.stress.mucin$Abundance / 11 *100


phylum.table.mucin <- rbind(phylum.table.mucin.baseline.3wk, phylum.table.stress)
phylum.table.mucin <- rbind(phylum.table.mucin, phylum.table.stress.mucin)


phylum.table.mucin <- as.data.frame(phylum.table.mucin)


# I'll plot the data to get an overview of differences between groups.
ggplot(phylum.table.mucin, aes(x = Group, y = Relative.Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") 

# This figure shows that the ratio of Bacteroidota to Firmicutes increases over time. However, I 
# want to clean up the figure by removing all Phyla that make up less than 1% of the total in any
# group and placing them in their own category.


# First, I'll summarize relative abundance information per taxonomic group.
summarized.abundance.phylum <- phylum.table.mucin %>%
  group_by(Phylum, Group) %>%
  summarize(Total.Relative.Abundance = (sum(Relative.Abundance)))

summarized.abundance.phylum$Phylum <- as.character(replace_na(summarized.abundance.phylum$Phylum, replace = "Unidentified"))

# Now, I'll collect taxa names for taxa with relative abundance above 1% in any group. These
# names will be preserved in each group.
summarized.abundance.phylum.above1 <- filter(summarized.abundance.phylum, Total.Relative.Abundance >= 1.0)
summarized.abundance.phylum.above1$Phylum <- factor(summarized.abundance.phylum.above1$Phylum, 
                                                    levels = sort(unique(summarized.abundance.phylum.above1$Phylum)))
abundant.families <- as.character(unique(summarized.abundance.phylum.above1$Phylum)) 
table(abundant.families) # There are 3 Phyla that have a Total Relative Abundance above 1%

# Make new data frames either including or excluding abundant families.
summarized.abundance.phylum.no.remainder <- filter(summarized.abundance.phylum, 
                                                   Phylum %in% abundant.families)

# I'll now summarize the remainder samples that compose less than 1% of relative abundance per group.
summarized.abundance.phylum.remainders.only <- anti_join(summarized.abundance.phylum, summarized.abundance.phylum.no.remainder, by = "Phylum")

summarized.abundance.phylum.remainders.only.summary <- summarized.abundance.phylum.remainders.only %>%
  group_by(Group) %>%
  summarize(Total.Relative.Abundance = (sum(Total.Relative.Abundance))) %>%
  mutate(Phylum = "Phyla < 1%", .before = 1)


# Now I'll combine abundant families with the low abundance remainders to account for 100% of total 
# abundance. This data frame will be used to generate the community composition figure.

summarized.abundance.phylum.figure <- rbind(summarized.abundance.phylum.no.remainder, summarized.abundance.phylum.remainders.only.summary)
summarized.abundance.phylum.figure$Phylum <- factor(summarized.abundance.phylum.figure$Phylum, levels = c("Bacteroidota", "Firmicutes", "Verrucomicrobiota", "Phyla < 1%"))

```
#### Phylum-level Community Composition
```{r echo = FALSE}
ggplot(summarized.abundance.phylum.figure, aes(x = Group, y = Total.Relative.Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  labs(title = "Community Composition (Phylum Level)", x = NULL, y = "Relative Abundance (%)") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
```
<br>

At the Phylum level, samples are dominated by members of the *Bacteroidota* and *Firmicutes* phyla. There is a general trend of relative decreased abundance in *Firmicutes* and relative increased abundance of *Bacteroidota* over the course of the experiment.
<br><br>

```{r include = FALSE}
ps.mucin.prop.family <- tax_glom(ps.mucin.prop, "Family", NArm = FALSE)
family.table.mucin <- psmelt(ps.mucin.prop.family) # Organize in long format for ggplot.

table(ps.mucin@sam_data$Group) # There are 24 samples for "Baseline" and "3 Wk Stress" groups, 
# 12 samples for the "Stress Alone" group, and 11 samples for the "Stress + Mucin" group. I'll 
# separate these groups to generate relative abundances as a percentage by dividing by the total
# number of samples in each group.

family.table.mucin.baseline.3wk <- filter(family.table.mucin, Group == "Baseline" | Group == "3 Wk Stress")
family.table.mucin.baseline.3wk$Relative.Abundance <- NA
family.table.mucin.baseline.3wk$Relative.Abundance <- family.table.mucin.baseline.3wk$Abundance / 24 *100


family.table.stress <- filter(family.table.mucin, Group == "Stress Alone")
family.table.stressRelative.Abundance <- NA
family.table.stress$Relative.Abundance <- family.table.stress$Abundance / 12 *100


family.table.stress.mucin <- filter(family.table.mucin, Group == "Stress + Mucin")
family.table.stress.mucinRelative.Abundance <- NA
family.table.stress.mucin$Relative.Abundance <- family.table.stress.mucin$Abundance / 11 *100


family.table.mucin <- rbind(family.table.mucin.baseline.3wk, family.table.stress)
family.table.mucin <- rbind(family.table.mucin, family.table.stress.mucin)


family.table.mucin <- as.data.frame(family.table.mucin)

# I'll plot the data to get an overview of differences between groups.
ggplot(family.table.mucin, aes(x = Group, y = Relative.Abundance, fill = Family)) +
  geom_bar(stat = "identity") 

# This figure is pretty cluttered. To clean it up, I'll filter out all Families that make up less than 1
# % of total in any group and place them in their own category.


# First, I'll summarize relative abundance information per taxonomic group.
summarized.abundance.family <- family.table.mucin %>%
  group_by(Family, Group) %>%
  summarize(Total.Relative.Abundance = (sum(Relative.Abundance)))

summarized.abundance.family$Family <- as.character(replace_na(summarized.abundance.family$Family, replace = "Unidentified"))

# Now, I'll collect taxa names for taxa with relative abundance above 1% in any group. These
# names will be preserved in each group.
summarized.abundance.family.above1 <- filter(summarized.abundance.family, Total.Relative.Abundance >= 1.0)
summarized.abundance.family.above1$Family <- factor(summarized.abundance.family.above1$Family, 
                                                    levels = sort(unique(summarized.abundance.family.above1$Family)))
abundant.families <- as.character(unique(summarized.abundance.family.above1$Family)) 
table(abundant.families) # There are 13 Families (and 1 group of unidentified families) that have a Total Relative 
# Abundance above 1%

# Make new data frames either including or excluding abundant families.
summarized.abundance.family.no.remainder <- filter(summarized.abundance.family, 
                                                   Family %in% abundant.families)

# I'll now summarize the remainder samples that compose less than 1% of relative abundance per group.
summarized.abundance.family.remainders.only <- anti_join(summarized.abundance.family, summarized.abundance.family.no.remainder, by = "Family")

summarized.abundance.family.remainders.only.summary <- summarized.abundance.family.remainders.only %>%
  group_by(Group) %>%
  summarize(Total.Relative.Abundance = (sum(Total.Relative.Abundance))) %>%
  mutate(Family = "Families < 1%", .before = 1)


# Now I'll combine abundant families with the low abundance remainders to account for 100% of total 
# abundance. This data frame will be used to generate the community composition figure.

summarized.abundance.family.figure <- rbind(summarized.abundance.family.no.remainder, summarized.abundance.family.remainders.only.summary)
summarized.abundance.family.figure$Family <- factor(summarized.abundance.family.figure$Family, levels = c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", 
                                                                                                          "Lactobacillaceae", "Marinifilaceae", "Muribaculaceae", "Oscillospiraceae", "Peptostreptococcaceae", "Rikenellaceae", "Ruminococcaceae", 
                                                                                                          "Unidentified", "Families < 1%"))
```
<br>

#### Family-Level Community Composition

```{r echo = FALSE}

ggplot(summarized.abundance.family.figure, aes(x = Group, y = Total.Relative.Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  labs(title = "Community Composition (Family Level)", x = NULL, y = "Relative Abundance (%)") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
```

Several Families are abundant across all groups, with the most abundant families including *Lachnospiraceae* and *Muribaculaceae*. I'll directly compare relative abundances of these families between groups of interest (Baseline vs.3 Week Stress and Stress Alone vs. Stress + Mucin).
<br>

#### Baseline vs. 3 Week Stress


```{r include = FALSE}
#### FAMILY LEVEL DIFFERENCES: BASELINE VS 3 WEEK STRESS ############################################
# I now want to plot these differences by Family group. I'll only be plotting the abundant families
# (those with a relative abundance above 1% in at least one Group). I'll start by taking the family
# table I generated previously and filtering it to include only these abundant families.

family.table.mucin.abundant <- filter(family.table.mucin, Family %in% abundant.families)

# I'll look at these comparisons in pairs to keep the figure from getting too complicated. I'll start
# by looking at the effect of stress alone, looking at Baseline vs. 3 week stress samples. I'll
# filter to include only these samples.

family.table.mucin.abundant.stress <- filter(family.table.mucin.abundant, Group == "Baseline" | Group == "3 Wk Stress")
family.table.mucin.abundant.stress$Group <- factor(family.table.mucin.abundant.stress$Group, levels = c("Baseline", "3 Wk Stress"))

family.table.mucin.abundant.stress <- filter(family.table.mucin.abundant.stress, Family != "Marinifilaceae")
family.table.mucin.abundant.stress <- filter(family.table.mucin.abundant.stress, Family != "Rikenellaceae")
family.table.mucin.abundant.stress$Family <- factor(family.table.mucin.abundant.stress$Family, levels = c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", 
                                                                                                          "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", 
                                                                                                          "Muribaculaceae", "Oscillospiraceae", "Peptostreptococcaceae", "Ruminococcaceae"))
```

```{r echo = FALSE}
ggplot(family.table.mucin.abundant.stress, aes(x = Abundance, y = Family, color = Group)) +
  geom_boxplot() +
  theme_bw() +
  scale_x_continuous(labels = scales::percent) +
  scale_y_discrete(limits = rev) +
  scale_color_manual(name = "Group", 
                     values = c("#941650", "#1c7798"), 
                     breaks = c("Baseline", "3 Wk Stress"), 
                     labels = c("Baseline", "3 Wk Stress")) +
  labs(title = "Family Abundance (Baseline vs. 3 Week Stresss)", x = "Relative Abundance", y = NULL) +
  theme(plot.title = element_text(hjust = 0.5))
```


``` {r include = FALSE}
### SIGNIFICANT DIFFERENCES: BASELINE VS. 3 WK STRESS ###
# I'll now use a Wilcoxon rank test to look for significant differences between each of these Family 
# groups. First, I'll organize the data so that abundance per Family is summarized on a per sample 
# basis. I'll also get Sample and Group metadata to add to summarized tables.
wilcox.family<- family.table.mucin.abundant.stress %>%
  group_by(Family, Sample) %>%
  summarize(Abundance.Group = (sum(Abundance)))

Sample.Group <- select(family.table.mucin.abundant.stress, Sample, Group)
Sample.Group <- Sample.Group %>%
  group_by(Sample) %>%
  slice(1L)

wilcox.family <- full_join(wilcox.family, Sample.Group, by = "Sample")

# Now I'll calculate Wilcoxon Ranked Sum p values for each Family.

### EUBACTERIA ###
wilcox.eubac <- filter(wilcox.family, Family == "[Eubacterium] coprostanoligenes group")

wilcox.eubac.baseline <- filter(wilcox.eubac, Group == "Baseline")
wilcox.eubac.3wk <- filter(wilcox.eubac, Group == "3 Wk Stress")

wilcox.test.eubac <- wilcox.test(wilcox.eubac.baseline$Abundance.Group, 
                                 wilcox.eubac.3wk$Abundance.Group)$p.value

### ACHOLEPLASMATACEAE ###
wilcox.achol <- filter(wilcox.family, Family == "Acholeplasmataceae")

wilcox.achol.baseline <- filter(wilcox.achol, Group == "Baseline")
wilcox.achol.3wk <- filter(wilcox.achol, Group == "3 Wk Stress")

wilcox.test.achol <- wilcox.test(wilcox.achol.baseline$Abundance.Group, 
                                 wilcox.achol.3wk$Abundance.Group)$p.value

### AKKERMANSIACEAE ###
wilcox.akker <- filter(wilcox.family, Family == "Akkermansiaceae")

wilcox.akker.baseline <- filter(wilcox.akker, Group == "Baseline")
wilcox.akker.3wk <- filter(wilcox.akker, Group == "3 Wk Stress")

wilcox.test.akker <- wilcox.test(wilcox.akker.baseline$Abundance.Group, 
                                 wilcox.akker.3wk$Abundance.Group)$p.value

### CLOSTRIDIACEAE ###
wilcox.clos <- filter(wilcox.family, Family == "Clostridiaceae")

wilcox.clos.baseline <- filter(wilcox.clos, Group == "Baseline")
wilcox.clos.3wk <- filter(wilcox.clos, Group == "3 Wk Stress")

wilcox.test.clos <- wilcox.test(wilcox.clos.baseline$Abundance.Group, 
                                wilcox.clos.3wk$Abundance.Group)$p.value

### ERYSIPELOTRICHACEAE ###
wilcox.erys <- filter(wilcox.family, Family == "Erysipelotrichaceae")

wilcox.erys.baseline <- filter(wilcox.erys, Group == "Baseline")
wilcox.erys.3wk <- filter(wilcox.erys, Group == "3 Wk Stress")

wilcox.test.erys <- wilcox.test(wilcox.erys.baseline$Abundance.Group, 
                                wilcox.erys.3wk$Abundance.Group)$p.value

### LACHNOSPIRACEAE ###
wilcox.lachno <- filter(wilcox.family, Family == "Lachnospiraceae")

wilcox.lachno.baseline <- filter(wilcox.lachno, Group == "Baseline")
wilcox.lachno.3wk <- filter(wilcox.lachno, Group == "3 Wk Stress")

wilcox.test.lachno <- wilcox.test(wilcox.lachno.baseline$Abundance.Group, 
                                wilcox.lachno.3wk$Abundance.Group)$p.value

### LACTOBACILLACEAE ###
wilcox.lacto <- filter(wilcox.family, Family == "Lactobacillaceae")

wilcox.lacto.baseline <- filter(wilcox.lacto, Group == "Baseline")
wilcox.lacto.3wk <- filter(wilcox.lacto, Group == "3 Wk Stress")

wilcox.test.lacto <- wilcox.test(wilcox.lacto.baseline$Abundance.Group, 
                                 wilcox.lacto.3wk$Abundance.Group)$p.value

### MURIBACULACEAE ###
wilcox.muri <- filter(wilcox.family, Family == "Muribaculaceae")

wilcox.muri.baseline <- filter(wilcox.muri, Group == "Baseline")
wilcox.muri.3wk <- filter(wilcox.muri, Group == "3 Wk Stress")

wilcox.test.muri <- wilcox.test(wilcox.muri.baseline$Abundance.Group, 
                                 wilcox.muri.3wk$Abundance.Group)$p.value

### OSCILLOSPIRACEAE ###
wilcox.oscil <- filter(wilcox.family, Family == "Oscillospiraceae")

wilcox.oscil.baseline <- filter(wilcox.oscil, Group == "Baseline")
wilcox.oscil.3wk <- filter(wilcox.oscil, Group == "3 Wk Stress")

wilcox.test.oscil <- wilcox.test(wilcox.oscil.baseline$Abundance.Group, 
                                wilcox.oscil.3wk$Abundance.Group)$p.value

### PEPTOSCTREPTOCOCCACEAE ###
wilcox.pepto <- filter(wilcox.family, Family == "Peptostreptococcaceae")

wilcox.pepto.baseline <- filter(wilcox.pepto, Group == "Baseline")
wilcox.pepto.3wk <- filter(wilcox.pepto, Group == "3 Wk Stress")

wilcox.test.pepto <- wilcox.test(wilcox.pepto.baseline$Abundance.Group, 
                                 wilcox.pepto.3wk$Abundance.Group)$p.value


### RUMINOCOCCACEAE ###
wilcox.rumin <- filter(wilcox.family, Family == "Ruminococcaceae")

wilcox.rumin.baseline <- filter(wilcox.rumin, Group == "Baseline")
wilcox.rumin.3wk <- filter(wilcox.rumin, Group == "3 Wk Stress")

wilcox.test.rumin <- wilcox.test(wilcox.rumin.baseline$Abundance.Group, 
                                 wilcox.rumin.3wk$Abundance.Group)$p.value


Wilcox.Families <- c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", 
                     "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", 
                     "Muribaculaceae", "Oscillospiraceae", "Peptostreptococcaceae", "Ruminococcaceae")

Wilcox.pvalues <- c(wilcox.test.eubac, wilcox.test.achol, wilcox.test.akker, wilcox.test.clos, wilcox.test.erys,
                    wilcox.test.lachno, wilcox.test.lacto, wilcox.test.muri, wilcox.test.oscil, wilcox.test.pepto, 
                    wilcox.test.rumin)

Wilcox.pvalues.Bonferonni <- p.adjust(Wilcox.pvalues, method = "bonferroni", n = 13)

Wilcox.Baseline.3wk <- cbind(Wilcox.Families, Wilcox.pvalues)
Wilcox.Baseline.3wk <- cbind(Wilcox.Baseline.3wk, Wilcox.pvalues.Bonferonni)
Wilcox.Baseline.3wk <- as.data.frame(Wilcox.Baseline.3wk)
Wilcox.Baseline.3wk <- plyr::rename(Wilcox.Baseline.3wk, replace = c("Wilcox.Families" = "Family",
                                                                     "Wilcox.pvalues" = "Raw p value",
                                                                     "Wilcox.pvalues.Bonferonni" = "Bonferroni-adjusted p value"))


#### FAMILY LEVEL DIFFERENCES: STRESS ALONE VS STRESS + MUCIN ########################################
# Now I'll repeat this process, comparing the 4 week samples. I want to see if there are any major
# differences in families between the Control group (Stress Alone) and the Mucin-supplemented group
# (Stress + Mucin). I'll filter to include only these samples.

family.table.mucin.abundant.mucin <- filter(family.table.mucin.abundant, Group == "Stress Alone" | Group == "Stress + Mucin")
family.table.mucin.abundant.mucin$Group <- factor(family.table.mucin.abundant.mucin$Group, levels = c("Stress Alone", "Stress + Mucin"))

family.table.mucin.abundant.mucin <- filter(family.table.mucin.abundant.mucin, Family != "Peptostreptococcaceae")
family.table.mucin.abundant.mucin$Family <- factor(family.table.mucin.abundant.mucin$Family, levels = c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", 
                                                                                                        "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", 
                                                                                                        "Marinifilaceae", "Muribaculaceae", "Oscillospiraceae", "Rikenellaceae", 
                                                                                                        "Ruminococcaceae"))
```

```{r echo = FALSE}
knitr::kable(Wilcox.Baseline.3wk)
```

Statistically significant differences were observed in the following families: *Eubacterium*, *Acholeplasmataceae*, *Clostridiaceae*, *Lactobacillaceae*, *Muribaculaceae*, *Peptostreptococcaceae*, and *Ruminococcaceae*. Of these Families, *Muribaculaceae* and *Lactobacillaceae* make up the largest proportion of the microbiota. Of note, *Lactobacillaceae* was lower in the stressed group relative to baseline, which is consistent with previous experiments.
<br>

#### Stress Alone vs. Stress + Mucin
```{r include = FALSE}
#### FAMILY LEVEL DIFFERENCES: STRESS ALONE VS STRESS + MUCIN ########################################
# Now I'll repeat this process, comparing the 4 week samples. I want to see if there are any major
# differences in families between the Control group (Stress Alone) and the Mucin-supplemented group
# (Stress + Mucin). I'll filter to include only these samples.

family.table.mucin.abundant.mucin <- filter(family.table.mucin.abundant, Group == "Stress Alone" | Group == "Stress + Mucin")
family.table.mucin.abundant.mucin$Group <- factor(family.table.mucin.abundant.mucin$Group, levels = c("Stress Alone", "Stress + Mucin"))

family.table.mucin.abundant.mucin <- filter(family.table.mucin.abundant.mucin, Family != "Peptostreptococcaceae")
family.table.mucin.abundant.mucin$Family <- factor(family.table.mucin.abundant.mucin$Family, levels = c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", 
                                                                                                        "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", 
                                                                                                        "Marinifilaceae", "Muribaculaceae", "Oscillospiraceae", "Rikenellaceae", 
                                                                                                        "Ruminococcaceae"))

```

```{r echo = FALSE}
ggplot(family.table.mucin.abundant.mucin, aes(x = Abundance, y = Family, color = Group)) +
  geom_boxplot() +
  theme_bw() +
  scale_x_continuous(labels = scales::percent) +
  scale_y_discrete(limits = rev) +
  scale_color_manual(name = "Group", 
                     values = c("#521b92", "#4d8e00"), 
                     breaks = c("Stress Alone", "Stress + Mucin"), 
                     labels = c("Stress Alone", "Stress + Mucin")) +
  labs(x = "Relative Abundance", y = NULL) 
```


```{r include = FALSE}
### SIGNIFICANT DIFFERENCES: STRESS ALONE VS. STRESS + MUCIN ###
# I'll now use a Wilcoxon rank test to look for significant differences between each of these Family 
# groups. First, I'll organize the data so that abundance per Family is summarized on a per sample 
# basis. I'll also get Sample and Group metadata to add to summarized tables.
wilcox.family<- family.table.mucin.abundant.mucin %>%
  group_by(Family, Sample) %>%
  summarize(Abundance.Group = (sum(Abundance)))

Sample.Group <- select(family.table.mucin.abundant.mucin, Sample, Group)
Sample.Group <- Sample.Group %>%
  group_by(Sample) %>%
  slice(1L)

wilcox.family <- full_join(wilcox.family, Sample.Group, by = "Sample")

# Now I'll calculate Wilcoxon Ranked Sum p values for each Family.
table(wilcox.eubac$Group)
### EUBACTERIA ###
wilcox.eubac <- filter(wilcox.family, Family == "[Eubacterium] coprostanoligenes group")

wilcox.eubac.control <- filter(wilcox.eubac, Group == "Stress Alone")
wilcox.eubac.mucin <- filter(wilcox.eubac, Group == "Stress + Mucin")

wilcox.test.eubac <- wilcox.test(wilcox.eubac.control$Abundance.Group, 
                                 wilcox.eubac.mucin$Abundance.Group)$p.value

### ACHOLEPLASMATACEAE ###
wilcox.achol <- filter(wilcox.family, Family == "Acholeplasmataceae")

wilcox.achol.control <- filter(wilcox.achol, Group == "Stress Alone")
wilcox.achol.mucin <- filter(wilcox.achol, Group == "Stress + Mucin")

wilcox.test.achol <- wilcox.test(wilcox.achol.control$Abundance.Group, 
                                 wilcox.achol.mucin$Abundance.Group)$p.value

### AKKERMANSIACEAE ###
wilcox.akker <- filter(wilcox.family, Family == "Akkermansiaceae")

wilcox.akker.control <- filter(wilcox.akker, Group == "Stress Alone")
wilcox.akker.mucin <- filter(wilcox.akker, Group == "Stress + Mucin")

wilcox.test.akker <- wilcox.test(wilcox.akker.control$Abundance.Group, 
                                 wilcox.akker.mucin$Abundance.Group)$p.value

### CLOSTRIDIACEAE ###
wilcox.clos <- filter(wilcox.family, Family == "Clostridiaceae")

wilcox.clos.control <- filter(wilcox.clos, Group == "Stress Alone")
wilcox.clos.mucin <- filter(wilcox.clos, Group == "Stress + Mucin")

wilcox.test.clos <- wilcox.test(wilcox.clos.control$Abundance.Group, 
                                wilcox.clos.mucin$Abundance.Group)$p.value

### ERYSIPELOTRICHACEAE ###
wilcox.erys <- filter(wilcox.family, Family == "Erysipelotrichaceae")

wilcox.erys.control <- filter(wilcox.erys, Group == "Stress Alone")
wilcox.erys.mucin <- filter(wilcox.erys, Group == "Stress + Mucin")

wilcox.test.erys <- wilcox.test(wilcox.erys.control$Abundance.Group, 
                                wilcox.erys.mucin$Abundance.Group)$p.value

### LACHNOSPIRACEAE ###
wilcox.lachno <- filter(wilcox.family, Family == "Lachnospiraceae")

wilcox.lachno.control <- filter(wilcox.lachno, Group == "Stress Alone")
wilcox.lachno.mucin <- filter(wilcox.lachno, Group == "Stress + Mucin")

wilcox.test.lachno <- wilcox.test(wilcox.lachno.control$Abundance.Group, 
                                  wilcox.lachno.mucin$Abundance.Group)$p.value

### LACTOBACILLACEAE ###
wilcox.lacto <- filter(wilcox.family, Family == "Lactobacillaceae")

wilcox.lacto.control <- filter(wilcox.lacto, Group == "Stress Alone")
wilcox.lacto.mucin <- filter(wilcox.lacto, Group == "Stress + Mucin")

wilcox.test.lacto <- wilcox.test(wilcox.lacto.control$Abundance.Group, 
                                 wilcox.lacto.mucin$Abundance.Group)$p.value

### MARINIFILACEAE ###
wilcox.marin <- filter(wilcox.family, Family == "Marinifilaceae")

wilcox.marin.control <- filter(wilcox.marin, Group == "Stress Alone")
wilcox.marin.mucin <- filter(wilcox.marin, Group == "Stress + Mucin")

wilcox.test.marin <- wilcox.test(wilcox.marin.control$Abundance.Group, 
                                 wilcox.marin.mucin$Abundance.Group)$p.value

### MURIBACULACEAE ###
wilcox.muri <- filter(wilcox.family, Family == "Muribaculaceae")

wilcox.muri.control <- filter(wilcox.muri, Group == "Stress Alone")
wilcox.muri.mucin <- filter(wilcox.muri, Group == "Stress + Mucin")

wilcox.test.muri <- wilcox.test(wilcox.muri.control$Abundance.Group, 
                                wilcox.muri.mucin$Abundance.Group)$p.value

### OSCILLOSPIRACEAE ###
wilcox.oscil <- filter(wilcox.family, Family == "Oscillospiraceae")

wilcox.oscil.control <- filter(wilcox.oscil, Group == "Stress Alone")
wilcox.oscil.mucin <- filter(wilcox.oscil, Group == "Stress + Mucin")

wilcox.test.oscil <- wilcox.test(wilcox.oscil.control$Abundance.Group, 
                                 wilcox.oscil.mucin$Abundance.Group)$p.value

### RIKENELLACEAE ###
wilcox.riken <- filter(wilcox.family, Family == "Rikenellaceae")

wilcox.riken.control <- filter(wilcox.riken, Group == "Stress Alone")
wilcox.riken.mucin <- filter(wilcox.riken, Group == "Stress + Mucin")

wilcox.test.riken <- wilcox.test(wilcox.riken.control$Abundance.Group, 
                                 wilcox.riken.mucin$Abundance.Group)$p.value

### RUMINOCOCCACEAE ###
wilcox.rumin <- filter(wilcox.family, Family == "Ruminococcaceae")

wilcox.rumin.control <- filter(wilcox.rumin, Group == "Stress Alone")
wilcox.rumin.mucin <- filter(wilcox.rumin, Group == "Stress + Mucin")

wilcox.test.rumin <- wilcox.test(wilcox.rumin.control$Abundance.Group, 
                                 wilcox.rumin.mucin$Abundance.Group)$p.value


Wilcox.Families <- c("[Eubacterium] coprostanoligenes group", "Acholeplasmataceae", "Akkermansiaceae", 
                     "Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", 
                     "Marinifilaceae", "Muribaculaceae", "Oscillospiraceae", "Rikenellaceae", 
                     "Ruminococcaceae")

Wilcox.pvalues <- c(wilcox.test.eubac, wilcox.test.achol, wilcox.test.akker, wilcox.test.clos, wilcox.test.erys,
                    wilcox.test.lachno, wilcox.test.lacto, wilcox.test.marin, wilcox.test.muri, wilcox.test.oscil,
                    wilcox.test.riken, wilcox.test.rumin)

Wilcox.pvalues.Bonferonni <- p.adjust(Wilcox.pvalues, method = "bonferroni", n = 12)

Wilcox.Control.Mucin <- cbind(Wilcox.Families, Wilcox.pvalues)
Wilcox.Control.Mucin <- cbind(Wilcox.Control.Mucin, Wilcox.pvalues.Bonferonni)
Wilcox.Control.Mucin <- as.data.frame(Wilcox.Control.Mucin)
Wilcox.Control.Mucin <- plyr::rename(Wilcox.Control.Mucin, replace = c("Wilcox.Families" = "Family",
                                                                     "Wilcox.pvalues" = "Unadjusted p value",
                                                                     "Wilcox.pvalues.Bonferonni" = "Bonferroni-adjusted p value"))
```

```{r echo = FALSE}
knitr::kable(Wilcox.Control.Mucin)
```

The *Rikenellaceae* and *Marinifilaceae* families were significantly different between Stress Alone and Stress + Mucin groups after correction for multiple comparisons. *Clostridiaceae* and *Lachnospiraceae* were different before adjustment for multiple comparisons. Of note, *Lactobacillaceae* was not significantly different between groups, indicating that Mucin supplementation is acting in a *Lactobacillaceae*-independent manner.

<br>

### Selection of ASVs that Distinguish Between Groups
I've broadly looked at differences in diversity and community composition with the Mucin Supplementation experiment for the Gautier TUMI Pilot. # I'll now move on to identify ASVs that best discriminate between groups. There are two comparisons we are interested in for this experiment:
## 1) Baseline (n=24) vs. 3 Wk Stress samples (n=24)
## 2) Stress only (n=12) vs. Stress + Mucin (n=11)

```{r include = FALSE}

```