---
title: "Stress Experiment Analysis Report"
author: "G. Brett Moreau"
date: "February 1, 2021"
output:
  html_document:
    toc: yes
---

``` {r setup, include = FALSE, warnings = FALSE}
#### PACKAGES ################################################
#if (!requireNamespace("BiocManager", quietly = TRUE))
#  install.packages("BiocManager")
#BiocManager::install()
packageVersion("BiocManager") # I'm using version 1.30.16

#BiocManager::install("phyloseq")
library(phyloseq)
packageVersion("phyloseq") # I'm using version 1.38.0

#install.packages("tidyverse")
library(tidyverse)
packageVersion("tidyverse") # I'm using version 1.3.1

#install.packages("ggplot2")
library(ggplot2)
packageVersion("ggplot2") # I'm using version 3.3.5

#BiocManager::install("microbiome")
library(microbiome) 
packageVersion("microbiome") # I'm using version 1.16.0

#install.packages("vegan")
library(vegan)
packageVersion("vegan") # I'm using version 2.5.7

#install.packages("caret")
library(caret)
packageVersion("caret") # I'm using version 6.0.90

#install.packages("randomForest")
library(randomForest)
packageVersion("randomForest") # I'm using version 4.6.14

#install.packages("pROC")
library(pROC)
packageVersion("pROC") # I'm using version 1.18.0
```
### Summary of Results

* There were no differences in alpha diversity (richness or evenness) between samples.

* Significant differences in beta (between samples) diversity were observed in samples Post-Stress compared to Pre-Stress treatment or Naive samples at the Post timepoint.

* Family-level community composition varied from sample to sample between all groups.

* Random Forest selected ASVs that distinguish between Pre- and Post-Stress groups. Several of these ASVs also distinguish between Baseline and 3 Week Stress groups from the Mucin experiment.

<br>

### Introduction
This analysis will focus on the Stress component of the Gautier TUMI Pilot project. This experiment focuses on the effect of stress on intestinal microbiota diversity and composition. Previous work in the lab has identified several changes in the microbiota, particularly decreases in *Lactobacillus ssp.*, that are associated with stress. This experiment will investigate this in a different set of samples to confirm reproducibility and identify other features associated with stress.

<br>

### Characterization of the Data Set
There are 24 samples in this experiment.Samples were divided into two groups (Naive and Stressed) and two different timepoints (pre- and post-stress intervention). Each treatment group and timepoint is divided into 6 samples each. A total of 421 Amplicon Sequence Variants (ASVs), or unique read sequences, were identified in these 24 samples.
<br>

Below is a general overview of the total number of reads in the data set, the mean and median number of reads/sample, and the range of reads across this data set.
<br>

```{r include = FALSE}
load(file = "/Users/gbm5pn/Documents/GitHub/Gautier-TUMI-Pilot-16S/results/phyloseq objects by experiment.RData")

# For this analysis, I'll be focusing on the Naive vs Stressed mice experiment (ps.stress), so I'll
# remove other objects from the environment.

rm(ps.mucin)

total.reads <- sample_sums(ps.stress)
```

```{r}
sum(total.reads) 
mean(total.reads) 
median(total.reads) 
range(total.reads) 
```

<br>

### Alpha Diversity
I'll first investigate alpha diversity, which examines diversity within a given sample. Alpha diversity consists of two major measures:

* **Richness**: the number of taxonomic groups within a given sample.

* **Evenness**: how evenly abundances of taxonomic groups are distributed in a given sample.
<br>

The DADA2 pipeline removes singlets (sequences detected only once) from the data set, as it does not have enough confidence to determine whether these sequences are real or artifacts. Because of this, measures of richness fail explicitly and are difficult to interpret. I'll focus mainly on measures of evenness but give measures of richness a glance to see if there are any major differences.
<br><br>

#### Richness
```{r include = FALSE}
ps.stress@sam_data$Condition <- factor(ps.stress@sam_data$Condition, 
                                       levels = c("Naïve Pre", "Stress Pre", "Naïve Post", "Stress Post"))
ps.stress.prop <- transform_sample_counts(ps.stress, function(ASV) ASV/sum(ASV)/6*100) 
```

```{r echo = FALSE}
plot_richness(ps.stress, x = "Condition", measures = c("Observed"), 
              color = "Condition") + 
  geom_point() +
  geom_boxplot() + 
  theme_bw() +
  theme(axis.title.x = element_blank()) +
  theme(axis.text.x = element_blank()) +
  scale_color_discrete(name = "Condition", 
                       breaks = c("Naïve Pre", "Stress Pre", "Naïve Post", "Stress Post"), 
                       labels = c("Naïve Pre", "Stress Pre", "Naïve Post", "Stress Post")) 

# Now I'll look for significant differences using a paired Wilcoxon test.
richness <- estimate_richness(ps.stress)
pairwise.wilcox.test(richness$Observed, p.adjust.method = "bonferroni", sample_data(ps.stress)$Condition)
```
<br>

There are no major differences in richness between groups.
<br><br>

#### Evenness
```{r include = FALSE}
sample.data.stress <- ps.stress@sam_data
sample.data.stress <- cbind("Sample.Names" = rownames(sample.data.stress), sample.data.stress)

evenness<- evenness(ps.stress, index = c("pielou"))
evenness<- cbind("Sample.Names" = rownames(evenness), evenness)

sample.data.stress <- full_join(sample.data.stress, evenness, by = "Sample.Names")

```

```{r echo = FALSE}
### PIELOU ###
ggplot(sample.data.stress, aes(x = Condition, y = pielou, color = Condition)) +
  geom_point() +
 geom_boxplot() +
  theme_bw() +
  labs(title = "Microbial Evenness Across Groups", x = NULL, y = "Pielou Eveness Index") +
  theme(legend.position = "none")

#ggsave("./results/figures/stress/Stress Experiment_evenness_pielou.png", width = 4, height = 4)

# Overall, there doesn't appear to be a whole lot of separation in evenness between groups. 
# There is a lot of spread between individuals within each group, which makes the data a 
# little difficult to interpret. Now I'll look for statistical differences.


pairwise.wilcox.test(evenness$pielou, p.adjust.method = "bonferroni", sample_data(ps.stress)$Condition)

```
<br>

There are also no major differences in evenness between groups. Overall, this suggests that stress does not drive differences in alpha diversity.

<br>

### Beta Diversity
I'll now investigate differences in beta diversity, or diversity between different samples in the experiment. I'll use Bray-Curtis Dissimilarity as my measure for beta diversity in this study, and I'll use Non-metric Multidimensional Scaling (NMDS) to visualize dissimilarity distances between samples.

```{r include = FALSE}
set.seed(763) # Set seed for reproducibility

# First, let's ordinate Bray-Curtis Dissimilarity using NMDS, then plot this ordination to visualize
# separation between different samples.

ord.nmds.bray <- ordinate(ps.stress.prop, method = "NMDS", distance = "bray")
```

```{r echo = FALSE}
plot_ordination(ps.stress.prop, ord.nmds.bray, color = "Condition", 
                title = "Bray-Curtis Dissimilarity") +
  theme_bw() +
  scale_color_discrete(name = "Condition", 
                       breaks = c("Naïve Pre", "Stress Pre", "Naïve Post", "Stress Post"), 
                       labels = c("Naïve Pre", "Stress Pre", "Naïve Post", "Stress Post")) +
  theme(aspect.ratio = 1, plot.title = element_text(hjust = 0.5)) 

```
<br>

Samples do not cluster into as clear of groups as seen with the Mucin supplementation experiment. However, there does appear to be some clustering berween groups. I'll perform a PERMANOVA to test for
differences in clustering between groups.

```{r include = FALSE}
adonis(distance(ps.stress.prop, method = "bray") ~sample_data(ps.stress.prop)$Condition)

# The PERMANOVA found significant differences in clustering (p = 0.001), but this doesn't specifiy
# which groups are significantly different from each other. To test this, I'll perform pairwise
# PERMANOVAs with multiple comparisons correction.

permanova.combinations <- combn(x = levels(ps.stress.prop@sam_data$Condition), m = 2)
permanova.p.bray <- vector(length = 6)
for(i in 1:ncol(permanova.combinations)){
  ps.subs.bray <- subset_samples(ps.stress.prop, Condition %in% permanova.combinations[,i])
  metadata.subs.bray <- as(sample_data(ps.subs.bray), "data.frame")
  pairwise.bray <- adonis(distance(ps.subs.bray, method = "bray") ~ Condition, data = metadata.subs.bray)
  permanova.p.bray[i] <- pairwise.bray$aov.tab[1,ncol(pairwise.bray$aov.tab)]
}
combinations <- c("Naïve Pre/Naïve Post", "Naïve Pre/Stress Pre", "Naïve Pre/Stress Post",
                  "Naïve Post/Stress Pre", "Naïve Post/Stress Post", "Stress Pre/Stress Post")

permanova.FDR.bray <- p.adjust(permanova.p.bray, method = "bonferroni")
permanova.bray.table <- cbind(combinations, permanova.p.bray)
permanova.bray.table <- cbind(permanova.bray.table, permanova.FDR.bray)
```

```{r echo = FALSE}
knitr::kable(permanova.bray.table)
```
<br>

Based on the pairwise comparisons, it looks like clustering is significantly different for all comaprisons except for Naive Pre vs. Naive Post and Naive Pre vs Stress Pre. Naive Pre and Post should be identical (no differences in treatment between these groups) and Naive Pre and Stress Pre should be identical (neither has received stress treatment), so these results are consistent with expectations. Overall, these data suggest that there are differences in the microbiota after stress treatment.

<br>

### Changes in Community Composition
There appear to be some broad differences in diversity between groups. I'll now see if this is reflected by any broad changes in community composition across groups. I'll start by looking at the phylum level, then move on to differences at the Family level. This analysis will focus on taxa that make up at least 1% of the relative abundance in at least one group.
<br><br>

#### Phylum-Level Community Composition
```{r include = FALSE}
### COMMUNITY COMPOSITION: PHYLUM LEVEL #############################################################
# There appear to be some broad differences in diversity between groups. I'll now see if this is
# reflected by any broad changes in community composition across groups. I'll start by looking at
# the phylum level, looking at all phyla that make up at least 1% of the relative abundance in at
# least one group.

ps.stress.prop.phylum <- tax_glom(ps.stress.prop, "Phylum", NArm = FALSE)
phylum.table.stress <- psmelt(ps.stress.prop.phylum) # Organize in long format for ggplot.

table(ps.stress.prop@sam_data$Condition)

# There are an equal number of samples (n=6) in each group. Because of this, the 
# Abundance column is already formatted to generate relative abundance as a percent
# of the total condition, so I'll just use this for analysis.

# I'll plot the data to get an overview of differences between groups.
ggplot(phylum.table.stress, aes(x = Condition, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") 

# I want to clean this figure up a little bit by including only phyla that make up
# at least 1% in at least one group.

# First, I'll summarize relative abundance information per taxonomic group.
summarized.abundance.phylum <- phylum.table.stress %>%
  group_by(Phylum, Condition) %>%
  summarize(Total.Relative.Abundance = (sum(Abundance)))

summarized.abundance.phylum$Phylum <- as.character(replace_na(summarized.abundance.phylum$Phylum, replace = "Unidentified"))

# Now, I'll collect taxa names for taxa with relative abundance above 1% in any group. These
# names will be preserved in each group.
summarized.abundance.phylum.above1 <- filter(summarized.abundance.phylum, Total.Relative.Abundance >= 1.0)
summarized.abundance.phylum.above1$Phylum <- factor(summarized.abundance.phylum.above1$Phylum, 
                                                    levels = sort(unique(summarized.abundance.phylum.above1$Phylum)))
abundant.families <- as.character(unique(summarized.abundance.phylum.above1$Phylum)) 
table(abundant.families) # There are 4 Phyla that have a Total Relative Abundance above 1%

# Make new data frames either including or excluding abundant families.
summarized.abundance.phylum.no.remainder <- filter(summarized.abundance.phylum, 
                                                   Phylum %in% abundant.families)

# I'll now summarize the remainder samples that compose less than 1% of relative abundance per group.
summarized.abundance.phylum.remainders.only <- anti_join(summarized.abundance.phylum, summarized.abundance.phylum.no.remainder, by = "Phylum")

summarized.abundance.phylum.remainders.only.summary <- summarized.abundance.phylum.remainders.only %>%
  group_by(Condition) %>%
  summarize(Total.Relative.Abundance = (sum(Total.Relative.Abundance))) %>%
  mutate(Phylum = "Phyla < 1%", .before = 1)


# Now I'll combine abundant families with the low abundance remainders to account for 100% of total 
# abundance. This data frame will be used to generate the community composition figure.

summarized.abundance.phylum.figure <- rbind(summarized.abundance.phylum.no.remainder, summarized.abundance.phylum.remainders.only.summary)
summarized.abundance.phylum.figure$Phylum <- factor(summarized.abundance.phylum.figure$Phylum, levels = c("Bacteroidota", "Firmicutes", "Proteobacteria", "Verrucomicrobiota", "Phyla < 1%"))
```

``` {r echo = FALSE}
ggplot(summarized.abundance.phylum.figure, aes(x = Condition, y = Total.Relative.Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  theme_bw() +
  labs(x = NULL, y = "Relative Abundance (%)")
```
<br>

At the Phylum level, samples were dominated by members of the *Bacteroidota* and *Firmicutes* phyla. Samples from the Naive group tended to have higher relative abundances of *Firmicutes* compared to samples from the Stressed Group at both timepoints.
<br><br>

#### Family-Level Community Composition
``` {r include = FALSE}
### COMMUNITY COMPOSITION: FAMILY LEVEL #############################################################
# There appear to be some differences at the phylum level. I'll now look at the family level to see
# how these differences are reflected here.

ps.stress.prop.family <- tax_glom(ps.stress.prop, "Family", NArm = FALSE)
family.table.stress <- psmelt(ps.stress.prop.family) # Organize in long format for ggplot.

table(ps.stress@sam_data$Condition) 

# Since all groups have equal number of samples, I'll just use the Abundance column.

# I'll plot the data to get an overview of differences between groups.
ggplot(family.table.stress, aes(x = Condition, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") 

# This figure is pretty cluttered. To clean it up, I'll filter out all Families that make up less than 1
# % of total in any group and place them in their own category.


# First, I'll summarize relative abundance information per taxonomic group.
summarized.abundance.family <- family.table.stress %>%
  group_by(Family, Condition) %>%
  summarize(Total.Relative.Abundance = (sum(Abundance)))

summarized.abundance.family$Family <- as.character(replace_na(summarized.abundance.family$Family, replace = "Unidentified"))

# Now, I'll collect taxa names for taxa with relative abundance above 1% in any group. These
# names will be preserved in each group.
summarized.abundance.family.above1 <- filter(summarized.abundance.family, Total.Relative.Abundance >= 1.0)
summarized.abundance.family.above1$Family <- factor(summarized.abundance.family.above1$Family, 
                                                    levels = sort(unique(summarized.abundance.family.above1$Family)))
abundant.families <- as.character(unique(summarized.abundance.family.above1$Family)) 
table(abundant.families) # There are 11 Families (and 1 group of unidentified families) that have a Total Relative 
# Abundance above 1%

# Make new data frames either including or excluding abundant families.
summarized.abundance.family.no.remainder <- filter(summarized.abundance.family, 
                                                   Family %in% abundant.families)

# I'll now summarize the remainder samples that compose less than 1% of relative abundance per group.
summarized.abundance.family.remainders.only <- anti_join(summarized.abundance.family, summarized.abundance.family.no.remainder, by = "Family")

summarized.abundance.family.remainders.only.summary <- summarized.abundance.family.remainders.only %>%
  group_by(Condition) %>%
  summarize(Total.Relative.Abundance = (sum(Total.Relative.Abundance))) %>%
  mutate(Family = "Families < 1%", .before = 1)


# Now I'll combine abundant families with the low abundance remainders to account for 100% of total 
# abundance. This data frame will be used to generate the community composition figure.

summarized.abundance.family.figure <- rbind(summarized.abundance.family.no.remainder, summarized.abundance.family.remainders.only.summary)
summarized.abundance.family.figure$Family <- factor(summarized.abundance.family.figure$Family, levels = c("Akkermansiaceae", "Bacteroidaceae","Clostridiaceae", "Erysipelotrichaceae", "Lachnospiraceae", "Lactobacillaceae", "Muribaculaceae", "Oscillospiraceae", "Peptostreptococcaceae", "Ruminococcaceae", "Sutterellaceae", "Unidentified", "Families < 1%"))                                                                                                   
  #### FAMILY LEVEL DIFFERENCES ########################################################################
# I now want to plot these differences by Family group. I'll only be plotting the abundant families
# (those with a relative abundance above 1% in at least one Condition).

family.table.stress.abundant <- filter(family.table.stress, Family %in% abundant.families)
family.table.stress.abundant$Abundance <- family.table.stress.abundant$Abundance *10

# I'll now plot abundance for each sample. Abundances are expressed as a percentage of each sample,
# so I'll format the x axis as a percentage.                                                                                                
```

```{r echo = FALSE}
ggplot(summarized.abundance.family.figure, aes(x = Condition, y = Total.Relative.Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  theme_bw() +
  labs(x = NULL, y = "Relative Abundance (%)")
```
<br><br>

Distribution of different Families appears to vary across different groups. I'll now look at the relative abundance of each family across groups to better visualize differences.

```{r echo = FALSE}
ggplot(family.table.stress.abundant, aes(x = Abundance, y = Family, color = Condition)) +
  geom_boxplot() +
  theme_bw() +
  scale_y_discrete(limits = rev) +
  labs(x = "Relative Abundance (%)", y = NULL) 
```

### Selection of ASVs that Distinguish Between Groups
I've broadly looked at differences in diversity and community composition with the Naive vs. Stress experiment. I'll now move on to identify ASVs that best discriminate between groups. For this analysis I'll focus on comparing the Stressed samples from Pre- and Post- conditions, as these are most comparable to the Baseline vs. 3 Week Stress samples from the Mucin supplementation experiment. ASVs will be selected by generating Random Forest models for each comparison. Because of the small sample size (n = 6 per group), I'll forego using training/test sets and build the Random Forest model on the entire data set.
<br>

```{r include = FALSE}
#### ORGANIZATION OF ASVs FROM STRESSED PRE VS POST SAMPLES #####################################
# I'll now look at differences in the Stressed mice Pre-stress vs Post-stress. 

ps.stress.ASV.time <- subset_samples(ps.stress, Condition == "Stress Pre" | Condition == "Stress Post")
ASV.table.stress.time <- psmelt(ps.stress.ASV.time)
View(ASV.table.stress.time)
length(unique(ASV.table.stress.time$OTU)) #1795 unique ASVs across all samples.

# I want to limit the data set to only ASVs that are present in at least one group. To do this, I'll
# first summarize the Abundance of each ASV in each group, then select all ASVs that are >0 in any
# group.

ASV.table.summary.stress.time <- ASV.table.stress.time %>%
  group_by(OTU, Condition) %>%
  summarize(Abundance.per.Condition = (sum(Abundance))) %>%
  filter(any(Abundance.per.Condition > 0)) # Removes any ASVs with an abundance of 0 in both groups.

length(unique(ASV.table.summary.stress.time$OTU)) # There are a total of 370 unique ASVs in these samples.


# Now I'll collect the ASV names for these ASVs. These will be used to merge only the ASV Abundance
# values for ASVs present in at least one group.
unique.ASVs.stress.time <- as.data.frame(unique(ASV.table.summary.stress.time$OTU))
unique.ASVs.stress.time <- plyr::rename(unique.ASVs.stress.time, replace = c("unique(ASV.table.summary.stress.time$OTU)" = "OTU"))

ASV.table.samples.stress.time <- left_join(unique.ASVs.stress.time, ASV.table.stress.time, by = "OTU")

length(unique(ASV.table.samples.stress.time$OTU)) # 370 unique ASVs as expected.


# I'll reshape the data into a wide format for analysis.
ASV.table.wide.stress.time <- select(ASV.table.samples.stress.time, OTU, Sample, Abundance, Condition, Animal.Number)
ASV.table.wide.stress.time <- spread(ASV.table.wide.stress.time, key = OTU, value = Abundance)
ASV.table.wide.stress.time$Condition <- factor(ASV.table.wide.stress.time$Condition, levels = c("Stress Pre", "Stress Post"))

View(ASV.table.wide.stress.time)

### RANDOM FOREST MODEL GENERATION ##################################################################
# There are 6 mice per group, which is a pretty small sample size. Because of this, I'll forego 
# using a training and test set and instead generate the random forest model on the entire data set.

# First, I'll separate the data into predictors and outcomes.
predictors.time <- select(ASV.table.wide.stress.time, -Sample, -Condition, -Animal.Number) # Remove metadata
outcome.time <- as.factor(ASV.table.wide.stress.time$Condition)

# The parameter I'll  tune for the random forest model is mtry, the number of features sampled 
# at each node of the decision tree. For this random forest, I'll use ntree (the number of trees in 
# the forest) = 1000, which is a pretty large number but not too computationally taxing. Then I'll 
# look at a variety of different mtry values to see which minimizes the Out-of-Bag error rate.

# By default mtry is set to the square root of the number of features. There are 370 features in 
# this data set, meaning mtry be default is set to 19. I'll test mtry values around 19 to see if 
# this improves model error.

# Set seed for reproducibility
#sample(1:1000, 1) # It selected 745
set.seed(745) 

### TUNING MTRY ###
model.rf.mtry15 <- randomForest(x = predictors.time, y = outcome.time, ntree = 1000, mtry = 15)
model.rf.mtry15 # The OOB error rate is 8.33%

model.rf.mtry17 <- randomForest(x = predictors.time, y = outcome.time, ntree = 1000, mtry = 17)
model.rf.mtry17 # The OOB error rate is 8.33%

model.rf.mtry19 <- randomForest(x = predictors.time, y = outcome.time, ntree = 1000, mtry = 19)
model.rf.mtry19 # The OOB error rate is 8.33%

model.rf.mtry21 <- randomForest(x = predictors.time, y = outcome.time, ntree = 1000, mtry = 21)
model.rf.mtry21 # The OOB error rate is 8.33%

model.rf.mtry23 <- randomForest(x = predictors.time, y = outcome.time, ntree = 1000, mtry = 23)
model.rf.mtry23 # The OOB error rate is 8.33%

# All random forest models performed the same, with an OOB error rate of 8.3%. This result was
# pretty consistent for all models, so I'll just go with the default mtry value of mtry == 19 
# for the final model.

model.rf.stress.time.final <- randomForest(x = predictors.time, y = outcome.time, ntree = 1000, mtry = 19)
model.rf.stress.time.final
```

```{r echo = FALSE}
roc(outcome.time, model.rf.stress.time.final$votes[,1], plot = TRUE, legacy.axes = TRUE, percent = TRUE, main = "ROC Curve",
    xlab = "False Positive Percentage", ylab = "True Positive Percentage", col = "#377eb8", lwd = 4, print.auc = TRUE)
```
<br>

This model performed well, but had less accuracy than previous models with larger sample sizes. The generated model had an Out-of-Bag error rate of 8.33%, and the overall AUC of the model is 83.3%. One caveat is that because the model was generated on the entire data set, we can't test it to see how well it predicts samples from a different data set.
<br>

```{r include = FALSE}
#### IMPORTANCE VALUES FROM RANDOM FOREST MODEL ####################################################
# Now that I've generated the model, I'll look at which features best discriminate between groups as
# selected by the model. I'll do this by checking the importance values from the Random Forest model, 
# using Gini index as the measure of importance.

ASV.taxonomy <- as.data.frame(ps.stress@tax_table)
ASV.taxonomy <- cbind(OTU = rownames(ASV.taxonomy), ASV.taxonomy)
rownames(ASV.taxonomy) <- NULL

importance.rf.stress.time <- importance(model.rf.stress.time.final, type = 2)

# Format the importance data as a data frame
importance.rf.stress.time <- cbind(OTU = rownames(importance.rf.stress.time), importance.rf.stress.time)
rownames(importance.rf.stress.time) <- NULL
importance.rf.stress.time <- as.data.frame(importance.rf.stress.time)
importance.rf.stress.time$MeanDecreaseGini <- as.numeric(importance.rf.stress.time$MeanDecreaseGini)

importance.rf.stress.time.taxonomy <- left_join(importance.rf.stress.time, ASV.taxonomy, by = "OTU")

# This is the table of ASVs listed by mean decrease in Gini Index. I'll rank the features according
# to Gini index to make the data easier to interpret.

RF.ranking.stress.time <- importance.rf.stress.time.taxonomy %>%
  arrange(desc(MeanDecreaseGini))

RF.ranking.stress.time <- plyr::rename(RF.ranking.stress.time, replace = c("MeanDecreaseGini" = "Pre.vs.Post.Mean.Decrease.Gini"))
RF.ranking.stress.time$Pre.Post.RF.Rank <- rank(-RF.ranking.stress.time$Pre.vs.Post.Mean.Decrease.Gini)

# I also want to add whether ASVs are enriched in the Naive or Stressed group.
ASV.table.enrichment.stress.time <- ASV.table.stress.time %>%
  group_by(OTU, Condition) %>%
  summarize(Average.Abundance.per.Group = (mean(Abundance))) %>%
  filter(any(Average.Abundance.per.Group > 0)) # Removes any ASVs with an abundance of 0 in both groups.

ASV.enrichment.Pre <- filter(ASV.table.enrichment.stress.time, Condition == "Stress Pre")
ASV.enrichment.Pre <- plyr::rename(ASV.enrichment.Pre, replace = c("Average.Abundance.per.Group" = "Pre.Average.Abundance.per.Group"))
ASV.enrichment.Pre <- select(ASV.enrichment.Pre, OTU, Pre.Average.Abundance.per.Group)

ASV.enrichment.Post <- filter(ASV.table.enrichment.stress.time, Condition == "Stress Post")
ASV.enrichment.Post <- plyr::rename(ASV.enrichment.Post, replace = c("Average.Abundance.per.Group" = "Post.Average.Abundance.per.Group"))
ASV.enrichment.Post <- select(ASV.enrichment.Post, OTU, 'Post.Average.Abundance.per.Group')

ASV.enrichment.stress.time <- full_join(ASV.enrichment.Pre, ASV.enrichment.Post, by = "OTU")
ASV.enrichment.stress.time$Enrichment.Pre.Post <- NA
ASV.enrichment.stress.time$Enrichment.Pre.Post[ASV.enrichment.stress.time$Pre.Average.Abundance.per.Group > ASV.enrichment.stress.time$Post.Average.Abundance.per.Group] <- "Stress Pre"
ASV.enrichment.stress.time$Enrichment.Pre.Post[ASV.enrichment.stress.time$Pre.Average.Abundance.per.Group < ASV.enrichment.stress.time$Post.Average.Abundance.per.Group] <- "Stress Post"

ASV.enrichment.time <- select(ASV.enrichment.stress.time, OTU, Enrichment.Pre.Post)

RF.ranking.stress.time <- full_join(RF.ranking.stress.time, ASV.enrichment.time, by = "OTU")

RF.ranking.stress.time$OTU <- factor(RF.ranking.stress.time$OTU, levels = RF.ranking.stress.time$OTU)

# I'll make new names for each figure, which will concatenate the ASV number and the Family
# designation for each ASV.
RF.ranking.stress.time$ASV.Family <- NA
RF.ranking.stress.time$ASV.Family <- paste(RF.ranking.stress.time$OTU, " (", RF.ranking.stress.time$Family, ")")
RF.ranking.stress.time$ASV.Family <- factor(RF.ranking.stress.time$ASV.Family, levels = RF.ranking.stress.time$ASV.Family)


ggplot(RF.ranking.stress.time, aes(x = Pre.vs.Post.Mean.Decrease.Gini, y = ASV.Family, color = Enrichment.Pre.Post)) +
  geom_point(size = 3) +
  labs(title = "Top Ranked Random Forest Predictors", x = "Mean Decrease in Node Impurity (Gini)", y = NULL) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"))

# To reduce the number of ASVs graphed on the figure, I'm going to set an arbitrarily cutoff 
# for the Gini Index to include only the most important ASVs.

RF.ranking.stress.time.figure <- filter(RF.ranking.stress.time, Pre.vs.Post.Mean.Decrease.Gini > 0.025)

```

```{r echo = FALSE}
ggplot(RF.ranking.stress.time.figure, aes(x = Pre.vs.Post.Mean.Decrease.Gini, y = ASV.Family, color = Enrichment.Pre.Post)) +
  geom_point(size = 3) +
  labs(title = "Top Ranked Random Forest Predictors", x = "Mean Decrease in Node Impurity (Gini)", y = NULL) +
  scale_color_manual(name = "Enrichment", 
                     values = c("#941650", "#1c7798"),
                     breaks = c("Stress Pre", "Stress Post"), 
                     labels = c("Stress Pre", "Stress Post")) +
  scale_y_discrete(limits=rev) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, face = "bold"))

```
<br>

Several ASVs were identified that distinguish between Pre- and Post-Stressed groups. Many of these ASVs were from the Family *Lachnospiraceae*, with many (but not all) of these ASVs enriched in the Post-Stress condition.

<br>

### Overlap Between ASVs From Stress and Mucin Experiments
While the Mucin supplementation experiment looks at the effect of mucin supplementation on stress, the "Baseline" and "3 Weeks" samples are identical to the "Stress Pre" and "Stress Post" groups in this experiment. Therefore, I'll look for overlap between the ASVs that discriminate in each experiment.

```{r include = FALSE}

# Read in data from mucin experiemnt.
RF.ranking.baseline.vs.stress <- read.csv("/Users/gbm5pn/Documents/GitHub/Gautier-TUMI-Pilot-16S/results/tables/mucin/RF Rankings_Baseline_vs_Stress.csv")


RF.ranking.stress.all <- select(RF.ranking.stress.time, OTU, Pre.vs.Post.Mean.Decrease.Gini, Pre.Post.RF.Rank, Enrichment.Pre.Post)

RF.ranking.stress.all <- full_join(RF.ranking.stress.all, RF.ranking.baseline.vs.stress, by = "OTU")
RF.ranking.stress.all <- select(RF.ranking.stress.all, OTU, Stress.Mean.Decrease.Gini, Pre.vs.Post.Mean.Decrease.Gini,
                                Stress.RF.Rank, Pre.Post.RF.Rank, Enrichment.Stress, Enrichment.Pre.Post, everything())

View(RF.ranking.stress.all)

# It doesn't look like there's a whole lot of overlap between data sets. To look for sure, I'll arbitrarily
# limit the data set to ASVs that were in the top 50 for each outcome by mean decrease in Gini Index.

RF.ranking.stress.all.important <- filter(RF.ranking.stress.all, Stress.RF.Rank < 50 & Pre.Post.RF.Rank < 50)                              

```

```{r echo = FALSE}
knitr::kable(RF.ranking.stress.all.important)
```
<br>

There are 10 ASVs that are ranked in the top 50 ASVs that discriminate between both comparisons (Baseline vs. 3 Week Stress for Mucin Experiment; Pre- vs. Post-Stress for Stress Experiment). All 10 ASVs are enriched consistently (i.e., enriched both at Baseline (Mucin) and Pre-Stress (Stress) or enriched both in 3 Weeks Stress(Mucin) and Post-Stress (Stress)). 6/10 of these ASVs are from Family *Lachnospiraceae*. Of these, 5/6 were enriched post-stress treatment. In addition, one of these ASVs is from Family *Lactobacillus*, which has been implicated previously in response to stress. In both experiments, the *Lactobacillus* ASV was enriched pre-stress, which is consistent with previous reports.

<br>

### Conclusions
Overall, there were no major differences in within-sample diversity (richness or evenness) between groups. Post-stress samples had significantly different beta diversity compared to Pre-stress samples or Naive samples at the Post timepoint. This suggests that stress treatment shifts microbiota diversity and is consistent with previous results as well as those seen in the Mucin supplementation experiment.
<br>

While there were shifts in Phylum- and Family-level community composition between groups, it was hard to distinguish true changes from noise. There seemed to be a lot of variation between samples. In addition, trends in the Family-level data were not necessarilly consistent with thsoe seen between Baseline and 3 Week Stress samples in the Mucin supplementation experiment. The limited number of samples for this experiment makes it difficult to draw conclusions. A Random Forest model was generated to identify ASVs that distinguish between groups and identified several ASVs, many of which from the *Lachnospiraceae* Family. Importantly, several highly ranked ASVs identified with this model were also identified by a Random Forest model distinguishing between Baseline and 3 Week Stress samples. Of the 10 ASVs identified by both models, all ASVs showed consistent associations (enriched in either the control or stressed condition in both), indicating that these ASVs may be conserved markers of changes in the microbiota due to stress.